diff -Naur a/meson_custom_install_desktop_file.sh b/meson_custom_install_desktop_file.sh
--- a/meson_custom_install_desktop_file.sh	2025-01-13 11:12:51.000000000 +0100
+++ b/meson_custom_install_desktop_file.sh	2025-01-14 23:01:28.692553386 +0100
@@ -2,7 +2,7 @@
 
 set -eu
 
-mkdir -p "${MESON_INSTALL_DESTDIR_PREFIX}/data/applications/"
+mkdir -p "${MESON_INSTALL_DESTDIR_PREFIX}/share/applications/"
 cp "${DESTDIR}/etc/xdg/autostart/user-dirs-update-gtk.desktop" \
-   "${MESON_INSTALL_DESTDIR_PREFIX}/data/applications/user-dirs-update-gtk.desktop"
+   "${MESON_INSTALL_DESTDIR_PREFIX}/share/applications/user-dirs-update-gtk.desktop"
 
