diff -Naur a/src/Fl_x.cxx b/src/Fl_x.cxx
--- a/src/Fl_x.cxx	2024-11-16 18:36:26.000000000 +0100
+++ b/src/Fl_x.cxx	2025-01-08 03:13:19.352323088 +0100
@@ -47,7 +47,7 @@
 #  include <X11/Xmd.h>
 #  include <X11/Xlocale.h>
 #  include <X11/Xlib.h>
-#  include <X11/keysym.h>
+#  include <X11/XKBlib.h>
 #  include "Xutf8.h"
 #define USE_XRANDR (HAVE_DLSYM && HAVE_DLFCN_H) // means attempt to dynamically load libXrandr.so
 #if USE_XRANDR
@@ -1465,40 +1465,6 @@
   return (long)total;
 }
 
-/*
-  Internal function to reduce "deprecated" warnings for XKeycodeToKeysym().
-  This way we get at most one warning. The option to use XkbKeycodeToKeysym()
-  instead would not help much - see STR #2913 for more information.
-
-  Update (July 22, 2022): disable "deprecated declaration" warnings in
-  this function for GCC >= 4.6 and clang (all versions) to get rid of
-  these warnings at least for current GCC and clang compilers.
-
-  Note: '#pragma GCC diagnostic push' needs at least GCC 4.6.
-*/
-
-#ifdef __clang__
-#pragma clang diagnostic push
-#pragma clang diagnostic ignored "-Wdeprecated-declarations"
-#endif
-
-#if defined(__GNUC__) && (__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ > 5))
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
-#endif
-
-static KeySym fl_KeycodeToKeysym(Display *d, KeyCode k, unsigned i) {
-  return XKeycodeToKeysym(d, k, i);
-}
-
-#if defined(__GNUC__) && (__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ > 5))
-#pragma GCC diagnostic pop
-#endif
-
-#ifdef __clang__
-#pragma clang diagnostic pop
-#endif
-
 static Window *xid_vector = NULL; // list of FLTK-created xid's (see issue #935)
 static int xid_vector_size = 0;
 static int xid_vector_count = 0;
@@ -2002,7 +1968,7 @@
 	  len = XUtf8LookupString(fl_xim_ic, (XKeyPressedEvent *)&xevent.xkey,
 			     kp_buffer, kp_buffer_len, &keysym, &status);
 	}
-	keysym = fl_KeycodeToKeysym(fl_display, keycode, 0);
+	keysym = XkbKeycodeToKeysym(fl_display, keycode, 0, 0);
       } else {
         //static XComposeStatus compose;
         len = XLookupString((XKeyEvent*)&(xevent.xkey),
@@ -2014,7 +1980,7 @@
           if (len < 1) len = 1;
           // ignore all effects of shift on the keysyms, which makes it a lot
           // easier to program shortcuts and is Windoze-compatible:
-          keysym = fl_KeycodeToKeysym(fl_display, keycode, 0);
+          keysym = XkbKeycodeToKeysym(fl_display, keycode, 0, 0);
         }
       }
       kp_buffer[len] = 0;
@@ -2057,7 +2023,7 @@
       event = FL_KEYUP;
       fl_key_vector[keycode/8] &= ~(1 << (keycode%8));
       // keyup events just get the unshifted keysym:
-      keysym = fl_KeycodeToKeysym(fl_display, keycode, 0);
+      keysym = XkbKeycodeToKeysym(fl_display, keycode, 0, 0);
     }
 #  ifdef __sgi
     // You can plug a microsoft keyboard into an sgi but the extra shift
@@ -2150,7 +2116,7 @@
     if (keysym >= 0xff91 && keysym <= 0xff9f) {
       // Map keypad keysym to character or keysym depending on
       // numlock state...
-      unsigned long keysym1 = fl_KeycodeToKeysym(fl_display, keycode, 1);
+      unsigned long keysym1 = XkbKeycodeToKeysym(fl_display, keycode, 1, 0);
       if (keysym1 <= 0x7f || (keysym1 > 0xff9f && keysym1 <= FL_KP_Last))
         Fl::e_original_keysym = (int)(keysym1 | FL_KP);
       if ((xevent.xkey.state & Mod2Mask) &&
