diff -Naur a/src/amd/llvm/ac_llvm_helper.cpp b/src/amd/llvm/ac_llvm_helper.cpp
--- a/src/amd/llvm/ac_llvm_helper.cpp	2023-09-28 23:11:25.000000000 +0200
+++ b/src/amd/llvm/ac_llvm_helper.cpp	2024-05-26 11:55:22.226422857 +0200
@@ -295,17 +299,12 @@
     */
    unwrap(passmgr)->add(createBarrierNoopPass());
 
-   /* This pass eliminates all loads and stores on alloca'd pointers. */
-   unwrap(passmgr)->add(createPromoteMemoryToRegisterPass());
    #if LLVM_VERSION_MAJOR >= 16
    unwrap(passmgr)->add(createSROAPass(true));
    #else
    unwrap(passmgr)->add(createSROAPass());
    #endif
    /* TODO: restore IPSCCP */
-   if (LLVM_VERSION_MAJOR >= 16)
-      unwrap(passmgr)->add(createLoopSinkPass());
-   /* TODO: restore IPSCCP */
    unwrap(passmgr)->add(createLICMPass());
    unwrap(passmgr)->add(createCFGSimplificationPass());
    /* This is recommended by the instruction combining pass. */
