diff -Naur a/resources/dist/linux/share/applications/org.moneymanagerex.MMEX.desktop b/resources/dist/linux/share/applications/org.moneymanagerex.MMEX.desktop
--- a/resources/dist/linux/share/applications/org.moneymanagerex.MMEX.desktop	2023-12-26 19:40:55.000000000 +0100
+++ b/resources/dist/linux/share/applications/org.moneymanagerex.MMEX.desktop	2024-01-27 19:29:43.347916476 +0100
@@ -137,28 +137,4 @@
 Terminal=false
 DocPath=mmex/help/index.html
 Categories=Finance;Office;
-Categories[cs]=Finance;Kancelář;
-Categories[de]=Finanzen;Büro;
-Categories[en]=Finance;Office;
-Categories[en_AU]=Finance;Office;
-Categories[en_GB]=Finance;Office;
-Categories[es]=Finanzas;Oficina;
-Categories[fi]=Rahoitus;Toimisto;
-Categories[fr]=Finances;Bureau;
-Categories[hu]=Pénzügy;Iroda;
-Categories[is]=Fjármál;Skrifstofa;
-Categories[it]=Finanza;Ufficio;
-Categories[kn]=ಹಣಕಾಸು;ಕಛೇರಿ;
-Categories[ko]=재원;사무실;
-Categories[lv]=Finanses;Birojs;
-Categories[nl]=Financiën;Kantoor;
-Categories[no_NY]=Financiën;Kantoor;
-Categories[pt]=Finanças;Escritório;
-Categories[pt_BR]=Finanças;Escritório;
-Categories[ro]=Finanțe;Birou;
-Categories[ru]=Финансы;Офис;
-Categories[sk]=Financie;Kancelária;
-Categories[sl]=Finance;Pisarna;
-Categories[tr]=Finans;Ofis;
-Categories[uk]=Фінанси;Офіс;
 MimeType=application/x-mmb;application/x-emb;application/x-mmdbg;
