diff -Naur a/Dbus/interfaces/vala/CMakeLists.txt b/Dbus/interfaces/vala/CMakeLists.txt
--- a/Dbus/interfaces/vala/CMakeLists.txt	2025-10-01 00:12:11.000000000 +0200
+++ b/Dbus/interfaces/vala/CMakeLists.txt	2025-10-03 19:58:20.967508828 +0200
@@ -62,5 +62,5 @@
 # now we can install vala files.
 add_subdirectory(src)
 # .pc goes with others
-configure_file (${CMAKE_CURRENT_SOURCE_DIR}/${CDAPPLET}.pc.in ${CMAKE_CURRENT_BINARY_DIR}/${CDAPPLET}.pc)
-install (FILES  ${CMAKE_CURRENT_BINARY_DIR}/${CDAPPLET}.pc DESTINATION "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}/pkgconfig")
+# configure_file (${CMAKE_CURRENT_SOURCE_DIR}/${CDAPPLET}.pc.in ${CMAKE_CURRENT_BINARY_DIR}/${CDAPPLET}.pc)
+# install (FILES  ${CMAKE_CURRENT_BINARY_DIR}/${CDAPPLET}.pc DESTINATION "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}/pkgconfig")
diff -Naur a/Dbus/interfaces/vala/src/CMakeLists.txt b/Dbus/interfaces/vala/src/CMakeLists.txt
--- a/Dbus/interfaces/vala/src/CMakeLists.txt	2025-10-01 00:12:11.000000000 +0200
+++ b/Dbus/interfaces/vala/src/CMakeLists.txt	2025-10-03 19:59:36.667695465 +0200
@@ -32,9 +32,9 @@
 
 
 # .so goes with other libs (to be found on runtime)
-install (TARGETS ${CDAPPLET} DESTINATION "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}")
+# install (TARGETS ${CDAPPLET} DESTINATION "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}")
 # .h goes in the dbus data dir
-install (FILES ${VALA_SRC_DIR}/${CDAPPLET}.h DESTINATION "${dbusdatadir}")
+# install (FILES ${VALA_SRC_DIR}/${CDAPPLET}.h DESTINATION "${dbusdatadir}")
 
 # Use the VAPI path from ValaVapiPath.txt file:
 find_program (HEAD_EXECUTABLE head)
@@ -43,5 +43,5 @@
 STRING (REGEX REPLACE "\n" "" VALA_VAPI_PATH ${VALA_VAPI_PATH})
 
 get_filename_component(VALA_VAPI_PATH "${VALA_SHARE_PATH}/${VALA_VAPI_PATH}" ABSOLUTE)
-install (FILES "${VALA_SRC_DIR}/${CDAPPLET}.vapi" DESTINATION "${VALA_VAPI_PATH}")
-install (FILES "${CMAKE_CURRENT_SOURCE_DIR}/../${CDAPPLET}.deps" DESTINATION "${VALA_VAPI_PATH}")
+# install (FILES "${VALA_SRC_DIR}/${CDAPPLET}.vapi" DESTINATION "${VALA_VAPI_PATH}")
+# install (FILES "${CMAKE_CURRENT_SOURCE_DIR}/../${CDAPPLET}.deps" DESTINATION "${VALA_VAPI_PATH}")
