diff -Naur a/external/redland/ExternalProject_raptor.mk b/external/redland/ExternalProject_raptor.mk
--- a/external/redland/ExternalProject_raptor.mk	2024-06-30 19:50:13.000000000 +0200
+++ b/external/redland/ExternalProject_raptor.mk	2024-07-12 00:49:09.057982021 +0200
@@ -30,7 +30,7 @@
 		CPPFLAGS="$(if $(SYSBASE),-I$(SYSBASE)/usr/include) $(gb_EMSCRIPTEN_CPPFLAGS)" \
 		$(gb_RUN_CONFIGURE) ./configure --disable-gtk-doc \
 			 --enable-parsers="rdfxml ntriples turtle trig guess rss-tag-soup" \
-			--with-www=xml \
+			--without-www \
 			--without-xslt-config \
 			$(gb_CONFIGURE_PLATFORMS) \
 			$(if $(CROSS_COMPILING),$(if $(filter INTEL ARM,$(CPUNAME)),ac_cv_c_bigendian=no)) \
