diff -Naur a/configure b/configure
--- a/configure	2025-03-21 16:39:03.000000000 +0100
+++ b/configure	2025-03-30 21:52:16.104791085 +0200
@@ -36005,19 +36005,19 @@
         SYSTEM_ICU=TRUE
 
 pkg_failed=no
-{ printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking for icu-i18n >= 66" >&5
-printf %s "checking for icu-i18n >= 66... " >&6; }
+{ printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking for icu-uc icu-i18n >= 66" >&5
+printf %s "checking for icu-uc icu-i18n >= 66... " >&6; }
 
 if test -n "$ICU_CFLAGS"; then
     pkg_cv_ICU_CFLAGS="$ICU_CFLAGS"
  elif test -n "$PKG_CONFIG"; then
     if test -n "$PKG_CONFIG" && \
-    { { printf "%s\n" "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"icu-i18n >= 66\""; } >&5
-  ($PKG_CONFIG --exists --print-errors "icu-i18n >= 66") 2>&5
+    { { printf "%s\n" "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"icu-uc icu-i18n >= 66\""; } >&5
+  ($PKG_CONFIG --exists --print-errors "icu-uc icu-i18n >= 66") 2>&5
   ac_status=$?
   printf "%s\n" "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
-  pkg_cv_ICU_CFLAGS=`$PKG_CONFIG --cflags "icu-i18n >= 66" 2>/dev/null`
+  pkg_cv_ICU_CFLAGS=`$PKG_CONFIG --cflags "icu-uc icu-i18n >= 66" 2>/dev/null`
 		      test "x$?" != "x0" && pkg_failed=yes
 else
   pkg_failed=yes
@@ -36029,12 +36029,12 @@
     pkg_cv_ICU_LIBS="$ICU_LIBS"
  elif test -n "$PKG_CONFIG"; then
     if test -n "$PKG_CONFIG" && \
-    { { printf "%s\n" "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"icu-i18n >= 66\""; } >&5
-  ($PKG_CONFIG --exists --print-errors "icu-i18n >= 66") 2>&5
+    { { printf "%s\n" "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"icu-uc icu-i18n >= 66\""; } >&5
+  ($PKG_CONFIG --exists --print-errors "icu-uc icu-i18n >= 66") 2>&5
   ac_status=$?
   printf "%s\n" "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
-  pkg_cv_ICU_LIBS=`$PKG_CONFIG --libs "icu-i18n >= 66" 2>/dev/null`
+  pkg_cv_ICU_LIBS=`$PKG_CONFIG --libs "icu-uc icu-i18n >= 66" 2>/dev/null`
 		      test "x$?" != "x0" && pkg_failed=yes
 else
   pkg_failed=yes
diff -Naur a/configure.ac b/configure.ac
--- a/configure.ac	2025-03-21 16:34:31.000000000 +0100
+++ b/configure.ac	2025-03-30 21:50:02.076237029 +0200
@@ -11312,14 +11312,14 @@
 ICU_MINOR=1
 ICU_CFLAGS_internal="-I${WORKDIR}/UnpackedTarball/icu/source/i18n -I${WORKDIR}/UnpackedTarball/icu/source/common"
 ICU_LIBS_internal="-L${WORKDIR}/UnpackedTarball/icu/source/lib -licuuc"
-libo_CHECK_SYSTEM_MODULE([icu],[ICU],[icu-i18n >= 66])
+libo_CHECK_SYSTEM_MODULE([icu],[ICU],[icu-uc icu-i18n >= 66])
 if test "$SYSTEM_ICU" = TRUE; then
     AC_LANG_PUSH([C++])
     AC_MSG_CHECKING([for unicode/rbbi.h])
     AC_PREPROC_IFELSE([AC_LANG_SOURCE([[unicode/rbbi.h]])],[AC_MSG_RESULT([yes])],[AC_MSG_ERROR([icu headers not found])])
     AC_LANG_POP([C++])
 
-    ICU_VERSION=`$PKG_CONFIG --modversion icu-i18n 2>/dev/null`
+    ICU_VERSION=`$PKG_CONFIG --modversion icu-uc icu-i18n 2>/dev/null`
     ICU_MAJOR=`echo $ICU_VERSION | cut -d"." -f1`
     ICU_MINOR=`echo $ICU_VERSION | cut -d"." -f2`
 
