diff -Naur a/dracut-functions.sh b/dracut-functions.sh
--- a/dracut-functions.sh	2022-06-20 00:35:26.000000000 +0200
+++ b/dracut-functions.sh	2024-04-15 22:18:48.926587635 +0200
@@ -709,8 +709,8 @@
     local _config_file
     [[ -f $dracutsysrootdir/boot/config-$kernel ]] \
         && _config_file="/boot/config-$kernel"
-    [[ -f $dracutsysrootdir/lib/modules/$kernel/config ]] \
-        && _config_file="/lib/modules/$kernel/config"
+    [[ -f $dracutsysrootdir/usr/lib/modules/$kernel/config ]] \
+        && _config_file="/usr/lib/modules/$kernel/config"
 
     # no kernel config file, so return true
     [[ $_config_file ]] || return 0
diff -Naur a/dracut-initramfs-restore.sh b/dracut-initramfs-restore.sh
--- a/dracut-initramfs-restore.sh	2022-06-20 00:35:26.000000000 +0200
+++ b/dracut-initramfs-restore.sh	2024-04-15 22:20:03.890865694 +0200
@@ -36,8 +36,8 @@
 elif [[ -d /boot/efi/loader/entries ]] || [[ -L /boot/efi/loader/entries ]] \
     || [[ -d /boot/efi/$MACHINE_ID ]] || [[ -L /boot/efi/$MACHINE_ID ]]; then
     IMG="/boot/efi/$MACHINE_ID/$KERNEL_VERSION/initrd"
-elif [[ -f /lib/modules/${KERNEL_VERSION}/initrd ]]; then
-    IMG="/lib/modules/${KERNEL_VERSION}/initrd"
+elif [[ -f /usr/lib/modules/${KERNEL_VERSION}/initrd ]]; then
+    IMG="/usr/lib/modules/${KERNEL_VERSION}/initrd"
 elif [[ -f /boot/initramfs-${KERNEL_VERSION}.img ]]; then
     IMG="/boot/initramfs-${KERNEL_VERSION}.img"
 elif mountpoint -q /efi; then
diff -Naur a/dracut-init.sh b/dracut-init.sh
--- a/dracut-init.sh	2022-06-20 00:35:26.000000000 +0200
+++ b/dracut-init.sh	2024-04-15 22:19:41.490782508 +0200
@@ -60,7 +60,7 @@
 fi
 
 # shellcheck disable=SC2154
-srcmods="$dracutsysrootdir/lib/modules/$kernel/"
+srcmods="$dracutsysrootdir/usr/lib/modules/$kernel/"
 
 # shellcheck disable=SC2154
 [[ $drivers_dir ]] && {
@@ -998,11 +998,11 @@
 
 dracut_kernel_post() {
     for _f in modules.builtin modules.builtin.alias modules.builtin.modinfo modules.order; do
-        [[ -e $srcmods/$_f ]] && inst_simple "$srcmods/$_f" "/lib/modules/$kernel/$_f"
+        [[ -e $srcmods/$_f ]] && inst_simple "$srcmods/$_f" "/usr/lib/modules/$kernel/$_f"
     done
 
     # generate module dependencies for the initrd
-    if [[ -d $initdir/lib/modules/$kernel ]] \
+    if [[ -d $initdir/usr/lib/modules/$kernel ]] \
         && ! depmod -a -b "$initdir" "$kernel"; then
         dfatal "\"depmod -a $kernel\" failed."
         exit 1
diff -Naur a/dracut.sh b/dracut.sh
--- a/dracut.sh	2022-06-20 00:35:26.000000000 +0200
+++ b/dracut.sh	2024-04-15 22:25:14.048025954 +0200
@@ -962,7 +962,7 @@
         esac
     done
 
-    cd "$dracutsysrootdir"/lib/modules || exit 1
+    cd "$dracutsysrootdir"/usr/lib/modules || exit 1
     if [[ $parallel != "yes" ]]; then
         for i in *; do
             [[ -f $i/modules.dep ]] || [[ -f $i/modules.dep.bin ]] || continue
@@ -1052,7 +1052,7 @@
 [[ $dracutbasedir ]] || dracutbasedir="$dracutsysrootdir"/usr/lib/dracut
 [[ $fw_dir ]] || {
     fw_path_para=$(< /sys/module/firmware_class/parameters/path)
-    fw_dir="${fw_path_para:+$dracutsysrootdir$fw_path_para:}$dracutsysrootdir/lib/firmware/updates/$kernel:$dracutsysrootdir/lib/firmware/updates:$dracutsysrootdir/lib/firmware/$kernel:$dracutsysrootdir/lib/firmware"
+    fw_dir="${fw_path_para:+$dracutsysrootdir$fw_path_para:}$dracutsysrootdir/usr/lib/firmware/updates/$kernel:$dracutsysrootdir/usr/lib/firmware/updates:$dracutsysrootdir/usr/lib/firmware/$kernel:$dracutsysrootdir/usr/lib/firmware"
 }
 [[ $tmpdir_l ]] && tmpdir="$tmpdir_l"
 [[ $tmpdir ]] || tmpdir="$TMPDIR"
@@ -1137,8 +1137,8 @@
             && [[ $MACHINE_ID ]] \
             && [[ -d "$dracutsysrootdir"/boot/efi/${MACHINE_ID} || -L "$dracutsysrootdir"/boot/efi/${MACHINE_ID} ]]; then
             outfile="$dracutsysrootdir/boot/efi/${MACHINE_ID}/${kernel}/initrd"
-        elif [[ -f "$dracutsysrootdir"/lib/modules/${kernel}/initrd ]]; then
-            outfile="$dracutsysrootdir/lib/modules/${kernel}/initrd"
+        elif [[ -f "$dracutsysrootdir"/usr/lib/modules/${kernel}/initrd ]]; then
+            outfile="$dracutsysrootdir/usr/lib/modules/${kernel}/initrd"
         elif [[ -e $dracutsysrootdir/boot/vmlinuz-${kernel} ]]; then
             outfile="$dracutsysrootdir/boot/initramfs-${kernel}.img"
         elif [[ -z $dracutsysrootdir ]] \
@@ -1211,7 +1211,7 @@
             printf "%s\n" 'dracut: -k/--kmoddir path must contain "lib/modules" as a parent of your kernel module directory,'
             printf "%s\n" "dracut: or modules may not be placed in the correct location inside the initramfs."
             printf "%s\n" "dracut: was given: ${drivers_dir}"
-            printf "%s\n" "dracut: expected: $(dirname "${drivers_dir}")/lib/modules/${kernel}"
+            printf "%s\n" "dracut: expected: $(dirname "${drivers_dir}")/usr/lib/modules/${kernel}"
             printf "%s\n" "dracut: Please move your modules into the correct directory structure and pass the new location,"
             printf "%s\n" "dracut: or set DRACUT_KMODDIR_OVERRIDE=1 to ignore this check."
             exit 1
@@ -1462,7 +1462,7 @@
         fi
 
         if ! [[ $kernel_image ]]; then
-            for kernel_image in "$dracutsysrootdir/lib/modules/$kernel/vmlinuz" "$dracutsysrootdir/boot/vmlinuz-$kernel"; do
+            for kernel_image in "$dracutsysrootdir/usr/lib/modules/$kernel/vmlinuz" "$dracutsysrootdir/boot/vmlinuz-$kernel"; do
                 [[ -s $kernel_image ]] || continue
                 break
             done
@@ -2109,7 +2109,7 @@
 
     if [[ $noimageifnotneeded == yes ]] && [[ $hostonly ]]; then
         if [[ ! -f "$initdir/lib/dracut/need-initqueue" ]] \
-            && [[ -f ${initdir}/lib/modules/$kernel/modules.dep && ! -s ${initdir}/lib/modules/$kernel/modules.dep ]]; then
+            && [[ -f ${initdir}/usr/lib/modules/$kernel/modules.dep && ! -s ${initdir}/usr/lib/modules/$kernel/modules.dep ]]; then
             for i in "${initdir}"/etc/cmdline.d/*.conf; do
                 # We need no initramfs image and do not generate one.
                 [[ $i == "${initdir}/etc/cmdline.d/*.conf" ]] && exit 0
diff -Naur a/lsinitrd.sh b/lsinitrd.sh
--- a/lsinitrd.sh	2022-06-20 00:35:26.000000000 +0200
+++ b/lsinitrd.sh	2024-04-15 22:20:25.486945974 +0200
@@ -129,8 +129,8 @@
         && [[ $MACHINE_ID ]] \
         && [[ -d /boot/efi/${MACHINE_ID} || -L /boot/efi/${MACHINE_ID} ]]; then
         image="/boot/efi/${MACHINE_ID}/${KERNEL_VERSION}/initrd"
-    elif [[ -f /lib/modules/${KERNEL_VERSION}/initrd ]]; then
-        image="/lib/modules/${KERNEL_VERSION}/initrd"
+    elif [[ -f /usr/lib/modules/${KERNEL_VERSION}/initrd ]]; then
+        image="/usr/lib/modules/${KERNEL_VERSION}/initrd"
     elif [[ -f /boot/initramfs-${KERNEL_VERSION}.img ]]; then
         image="/boot/initramfs-${KERNEL_VERSION}.img"
     elif [[ $MACHINE_ID ]] \
diff -Naur a/man/dracut.8.asc b/man/dracut.8.asc
--- a/man/dracut.8.asc	2022-06-20 00:35:26.000000000 +0200
+++ b/man/dracut.8.asc	2024-04-15 22:21:13.203123613 +0200
@@ -23,7 +23,7 @@
 _/efi/<machine-id>/<kernel-version>/initrd_,
 _/boot/<machine-id>/<kernel-version>/initrd_,
 _/boot/efi/<machine-id>/<kernel-version>/initrd_,
-_/lib/modules/<kernel-version>/initrd_ or
+_/usr/lib/modules/<kernel-version>/initrd_ or
 _/boot/initramfs-<kernel-version>.img_.
 
 dracut creates an initial image used by the kernel for preloading the block
@@ -597,7 +597,7 @@
 
 **--kernel-image _<file>_**::
     Specifies the kernel image, which to include in the UEFI executable. The
-    default is _/lib/modules/<KERNEL-VERSION>/vmlinuz_ or
+    default is _/usr/lib/modules/<KERNEL-VERSION>/vmlinuz_ or
     _/boot/vmlinuz-<KERNEL-VERSION>_.
 
 **--enhanced-cpio**::
diff -Naur a/man/dracut.conf.5.asc b/man/dracut.conf.5.asc
--- a/man/dracut.conf.5.asc	2022-06-20 00:35:26.000000000 +0200
+++ b/man/dracut.conf.5.asc	2024-04-15 22:21:43.115235169 +0200
@@ -296,7 +296,7 @@
 
 *kernel_image=*"_<file>_"::
     Specifies the kernel image, which to include in the UEFI executable. The
-    default is _/lib/modules/<KERNEL-VERSION>/vmlinuz_ or
+    default is _/usr/lib/modules/<KERNEL-VERSION>/vmlinuz_ or
     _/boot/vmlinuz-<KERNEL-VERSION>_.
 
 *enhanced_cpio=*"__{yes|no}__"::
diff -Naur a/man/dracut.usage.asc b/man/dracut.usage.asc
--- a/man/dracut.usage.asc	2022-06-20 00:35:26.000000000 +0200
+++ b/man/dracut.usage.asc	2024-04-15 22:20:50.435038803 +0200
@@ -9,7 +9,7 @@
 _/efi/_++<machine-id>++_/_++<kernel-version>++_/initrd_,
 _/boot/_++<machine-id>++_/_++<kernel-version>++_/initrd_,
 _/boot/efi/_++<machine-id>++_/_++<kernel-version>++_/initrd_,
-_/lib/modules/_++<kernel-version>++_/initrd_ or
+_/usr/lib/modules/_++<kernel-version>++_/initrd_ or
 _/boot/initramfs-_++<kernel-version>++_.img_ and contains the kernel modules of
 the currently active kernel with version _++<kernel-version>++_.
 
diff -Naur a/man/lsinitrd.1.asc b/man/lsinitrd.1.asc
--- a/man/lsinitrd.1.asc	2022-06-20 00:35:26.000000000 +0200
+++ b/man/lsinitrd.1.asc	2024-04-15 22:21:31.039190112 +0200
@@ -21,7 +21,7 @@
 lsinitrd uses the default image _/efi/<machine-id>/<kernel-version>/initrd_,
 _/boot/<machine-id>/<kernel-version>/initrd_,
 _/boot/efi/<machine-id>/<kernel-version>/initrd_,
-_/lib/modules/<kernel-version>/initrd_ or
+_/usr/lib/modules/<kernel-version>/initrd_ or
 _/boot/initramfs-<kernel-version>.img_.
 
 OPTIONS
diff -Naur a/modules.d/03modsign/load-modsign-keys.sh b/modules.d/03modsign/load-modsign-keys.sh
--- a/modules.d/03modsign/load-modsign-keys.sh	2022-06-20 00:35:26.000000000 +0200
+++ b/modules.d/03modsign/load-modsign-keys.sh	2024-04-15 22:23:20.979601162 +0200
@@ -5,7 +5,7 @@
 # Copyright 2013 Red Hat, Inc.
 # Peter Jones <pjones@redhat.com>
 
-for x in /lib/modules/keys/*; do
-    [ "${x}" = "/lib/modules/keys/*" ] && break
+for x in /usr/lib/modules/keys/*; do
+    [ "${x}" = "/usr/lib/modules/keys/*" ] && break
     keyctl padd asymmetric "" @s < "${x}"
 done
diff -Naur a/modules.d/03modsign/module-setup.sh b/modules.d/03modsign/module-setup.sh
--- a/modules.d/03modsign/module-setup.sh	2022-06-20 00:35:26.000000000 +0200
+++ b/modules.d/03modsign/module-setup.sh	2024-04-15 22:24:06.551772129 +0200
@@ -12,8 +12,8 @@
     # do not include module in hostonly mode,
     # if no keys are present
     if [[ $hostonly ]]; then
-        x=$(echo "$dracutsysrootdir"/lib/modules/keys/*)
-        [[ ${x} == "$dracutsysrootdir/lib/modules/keys/*" ]] && return 255
+        x=$(echo "$dracutsysrootdir"/usr/lib/modules/keys/*)
+        [[ ${x} == "$dracutsysrootdir/usr/lib/modules/keys/*" ]] && return 255
     fi
 
     return 0
@@ -26,13 +26,13 @@
 
 # called by dracut
 install() {
-    inst_dir /lib/modules/keys
+    inst_dir /usr/lib/modules/keys
     inst_binary keyctl
 
     inst_hook pre-trigger 01 "$moddir/load-modsign-keys.sh"
 
-    for x in "$dracutsysrootdir"/lib/modules/keys/*; do
-        [[ ${x} == "$dracutsysrootdir/lib/modules/keys/*" ]] && break
+    for x in "$dracutsysrootdir"/usr/lib/modules/keys/*; do
+        [[ ${x} == "$dracutsysrootdir/usr/lib/modules/keys/*" ]] && break
         inst_simple "${x#$dracutsysrootdir}"
     done
 }
diff -Naur a/modules.d/62bluetooth/module-setup.sh b/modules.d/62bluetooth/module-setup.sh
--- a/modules.d/62bluetooth/module-setup.sh	2022-06-20 00:35:26.000000000 +0200
+++ b/modules.d/62bluetooth/module-setup.sh	2024-04-15 22:30:28.241197982 +0200
@@ -28,23 +28,23 @@
 installkernel() {
     instmods bluetooth btrtl btintel btbcm bnep ath3k btusb rfcomm hidp
     inst_multiple -o \
-        /lib/firmware/ar3k/AthrBT* \
-        /lib/firmware/ar3k/ramps* \
-        /lib/firmware/ath3k-1.fw* \
-        /lib/firmware/BCM2033-MD.hex* \
-        /lib/firmware/bfubase.frm* \
-        /lib/firmware/BT3CPCC.bin* \
-        /lib/firmware/brcm/*.hcd* \
-        /lib/firmware/mediatek/mt7622pr2h.bin* \
-        /lib/firmware/qca/nvm* \
-        /lib/firmware/qca/crnv* \
-        /lib/firmware/qca/rampatch* \
-        /lib/firmware/qca/crbtfw* \
-        /lib/firmware/rtl_bt/* \
-        /lib/firmware/intel/ibt* \
-        /lib/firmware/ti-connectivity/TIInit_* \
-        /lib/firmware/nokia/bcmfw.bin* \
-        /lib/firmware/nokia/ti1273.bin*
+        /usr/lib/firmware/ar3k/AthrBT* \
+        /usr/lib/firmware/ar3k/ramps* \
+        /usr/lib/firmware/ath3k-1.fw* \
+        /usr/lib/firmware/BCM2033-MD.hex* \
+        /usr/lib/firmware/bfubase.frm* \
+        /usr/lib/firmware/BT3CPCC.bin* \
+        /usr/lib/firmware/brcm/*.hcd* \
+        /usr/lib/firmware/mediatek/mt7622pr2h.bin* \
+        /usr/lib/firmware/qca/nvm* \
+        /usr/lib/firmware/qca/crnv* \
+        /usr/lib/firmware/qca/rampatch* \
+        /usr/lib/firmware/qca/crbtfw* \
+        /usr/lib/firmware/rtl_bt/* \
+        /usr/lib/firmware/intel/ibt* \
+        /usr/lib/firmware/ti-connectivity/TIInit_* \
+        /usr/lib/firmware/nokia/bcmfw.bin* \
+        /usr/lib/firmware/nokia/ti1273.bin*
 }
 
 # Install the required file(s) for the module in the initramfs.
diff -Naur a/shell-completion/bash/dracut b/shell-completion/bash/dracut
--- a/shell-completion/bash/dracut	2022-06-20 00:35:26.000000000 +0200
+++ b/shell-completion/bash/dracut	2024-04-15 22:22:51.455490586 +0200
@@ -73,7 +73,7 @@
                 ;;
             --kver)
                 comps=$(
-                    cd /lib/modules || return 0
+                    cd /usr/lib/modules || return 0
                     echo [0-9]*
                 )
                 ;;
@@ -111,7 +111,7 @@
     elif [[ $args -eq 2 ]]; then
         # shellcheck disable=SC2034
         comps=$(
-            cd /lib/modules || return 0
+            cd /usr/lib/modules || return 0
             echo [0-9]*
         )
         # shellcheck disable=SC2207
diff -Naur a/shell-completion/bash/lsinitrd b/shell-completion/bash/lsinitrd
--- a/shell-completion/bash/lsinitrd	2022-06-20 00:35:26.000000000 +0200
+++ b/shell-completion/bash/lsinitrd	2024-04-15 22:23:07.487550614 +0200
@@ -38,7 +38,7 @@
                 ;;
             --kver | -k)
                 comps=$(
-                    cd /lib/modules || return 1
+                    cd /usr/lib/modules || return 1
                     echo [0-9]*
                 )
                 ;;
diff -Naur a/src/install/dracut-install.c b/src/install/dracut-install.c
--- a/src/install/dracut-install.c	2022-06-20 00:35:26.000000000 +0200
+++ b/src/install/dracut-install.c	2024-04-15 22:26:13.880251555 +0200
@@ -983,12 +983,12 @@
                "\n"
                "  --module,-m       Install kernel modules, instead of files\n"
                "  --kerneldir       Specify the kernel module directory\n"
-               "                     (default: /lib/modules/$(uname -r))\n"
+               "                     (default: /usr/lib/modules/$(uname -r))\n"
                "  --firmwaredirs    Specify the firmware directory search path with : separation\n"
                "                     (default: $DRACUT_FIRMWARE_PATH, otherwise kernel-compatible\n"
                "                      $(</sys/module/firmware_class/parameters/path),\n"
-               "                      /lib/firmware/updates/$(uname -r), /lib/firmware/updates\n"
-               "                      /lib/firmware/$(uname -r), /lib/firmware)\n"
+               "                      /usr/lib/firmware/updates/$(uname -r), /usr/lib/firmware/updates\n"
+               "                      /usr/lib/firmware/$(uname -r), /usr/lib/firmware)\n"
                "  --silent          Don't display error messages for kernel module install\n"
                "  --modalias        Only generate module list from /sys/devices modalias list\n"
                "  -o --optional     If kernel module does not exist, do not fail\n"
@@ -1155,7 +1155,7 @@
         struct utsname buf = {0};
         if (!kerneldir) {
                 uname(&buf);
-                _asprintf(&kerneldir, "/lib/modules/%s", buf.release);
+                _asprintf(&kerneldir, "/usr/lib/modules/%s", buf.release);
         }
 
         if (arg_modalias) {
@@ -1182,13 +1182,13 @@
                                         close(path);
                                 }
                                 char uk[22 + sizeof(buf.release)], fk[14 + sizeof(buf.release)];
-                                sprintf(uk, "/lib/firmware/updates/%s", buf.release);
-                                sprintf(fk, "/lib/firmware/%s", buf.release);
+                                sprintf(uk, "/usr/lib/firmware/updates/%s", buf.release);
+                                sprintf(fk, "/usr/lib/firmware/%s", buf.release);
                                 firmwaredirs = strv_new(STRV_IFNOTNULL(*fw_path_para ? fw_path_para : NULL),
                                                         uk,
-                                                        "/lib/firmware/updates",
+                                                        "/usr/lib/firmware/updates",
                                                         fk,
-                                                        "/lib/firmware",
+                                                        "/usr/lib/firmware",
                                                         NULL);
                         }
                 }
@@ -1723,7 +1723,7 @@
         ctx = kmod_new(kerneldir, NULL);
         abskpath = kmod_get_dirname(ctx);
 
-        p = strstr(abskpath, "/lib/modules/");
+        p = strstr(abskpath, "/usr/lib/modules/");
         if (p != NULL)
                 kerneldirlen = p - abskpath;
 
