diff -Naur a/src/tools.c b/src/tools.c
--- a/src/tools.c	2021-01-14 02:00:00.000000000 +0100
+++ b/src/tools.c	2025-04-27 17:11:13.723866940 +0200
@@ -35,7 +35,7 @@
               gpointer     data)
 {
     egg_sm_client_will_quit (client, TRUE);
-    gnome_encfs_manager_on_logout();
+//    gnome_encfs_manager_on_logout();
 }
 
 /* quit handler for the master client */
@@ -43,7 +43,7 @@
 client_quit_cb (EggSMClient *client,
         gpointer     data)
 {
-    gnome_encfs_manager_on_logout();
+//    gnome_encfs_manager_on_logout();
     gtk_main_quit ();
 }
 
