diff -Naur a/cmake/libprotobuf.cmake b/cmake/libprotobuf.cmake
--- a/cmake/libprotobuf.cmake	2024-03-12 19:01:14.000000000 +0100
+++ b/cmake/libprotobuf.cmake	2024-03-15 00:56:47.905999027 +0100
@@ -36,6 +36,7 @@
 endif()
 set_target_properties(libprotobuf PROPERTIES
     VERSION ${protobuf_VERSION}
+    SOVERSION ${protobuf_VERSION_MINOR}
     OUTPUT_NAME ${LIB_PREFIX}protobuf
     DEBUG_POSTFIX "${protobuf_DEBUG_POSTFIX}"
     # For -fvisibility=hidden and -fvisibility-inlines-hidden
diff -Naur a/cmake/libprotobuf-lite.cmake b/cmake/libprotobuf-lite.cmake
--- a/cmake/libprotobuf-lite.cmake	2024-03-12 19:01:14.000000000 +0100
+++ b/cmake/libprotobuf-lite.cmake	2024-03-15 00:56:37.105963762 +0100
@@ -33,6 +33,7 @@
 endif()
 set_target_properties(libprotobuf-lite PROPERTIES
     VERSION ${protobuf_VERSION}
+    SOVERSION ${protobuf_VERSION_MINOR}
     OUTPUT_NAME ${LIB_PREFIX}protobuf-lite
     DEBUG_POSTFIX "${protobuf_DEBUG_POSTFIX}"
     # For -fvisibility=hidden and -fvisibility-inlines-hidden
diff -Naur a/cmake/libprotoc.cmake b/cmake/libprotoc.cmake
--- a/cmake/libprotoc.cmake	2024-03-12 19:01:14.000000000 +0100
+++ b/cmake/libprotoc.cmake	2024-03-15 00:56:59.106035587 +0100
@@ -27,6 +27,7 @@
 set_target_properties(libprotoc PROPERTIES
     COMPILE_DEFINITIONS LIBPROTOC_EXPORTS
     VERSION ${protobuf_VERSION}
+    SOVERSION ${protobuf_VERSION_MINOR}
     OUTPUT_NAME ${LIB_PREFIX}protoc
     DEBUG_POSTFIX "${protobuf_DEBUG_POSTFIX}"
     # For -fvisibility=hidden and -fvisibility-inlines-hidden
