diff -Naur a/aclocal.m4 b/aclocal.m4
--- a/aclocal.m4	2023-02-01 04:27:45.000000000 +0100
+++ b/aclocal.m4	2023-04-06 21:36:32.903047509 +0200
@@ -306,10 +306,10 @@
 [test -n "$libc_cv_slibdir" ||
 case "$prefix" in
 /usr | /usr/)
-  libc_cv_slibdir='/$1'
-  libc_cv_rtlddir='/$2'
+  libc_cv_slibdir='/lib'
+  libc_cv_rtlddir='/lib'
   if test "$libdir" = '${exec_prefix}/lib'; then
-    libdir='${exec_prefix}/$1';
+    libdir='${exec_prefix}/lib';
     # Locale data can be shared between 32-bit and 64-bit libraries.
     libc_cv_complocaledir='${exec_prefix}/lib/locale'
   fi
diff -Naur a/sysdeps/unix/sysv/linux/x86_64/64/configure b/sysdeps/unix/sysv/linux/x86_64/64/configure
--- a/sysdeps/unix/sysv/linux/x86_64/64/configure	2023-02-01 04:27:45.000000000 +0100
+++ b/sysdeps/unix/sysv/linux/x86_64/64/configure	2023-04-06 21:34:40.942816519 +0200
@@ -4,10 +4,10 @@
 test -n "$libc_cv_slibdir" ||
 case "$prefix" in
 /usr | /usr/)
-  libc_cv_slibdir='/lib64'
-  libc_cv_rtlddir='/lib64'
+  libc_cv_slibdir='/lib'
+  libc_cv_rtlddir='/lib'
   if test "$libdir" = '${exec_prefix}/lib'; then
-    libdir='${exec_prefix}/lib64';
+    libdir='${exec_prefix}/lib';
     # Locale data can be shared between 32-bit and 64-bit libraries.
     libc_cv_complocaledir='${exec_prefix}/lib/locale'
   fi
diff -Naur a/sysdeps/unix/sysv/linux/x86_64/64/configure.ac b/sysdeps/unix/sysv/linux/x86_64/64/configure.ac
--- a/sysdeps/unix/sysv/linux/x86_64/64/configure.ac	2023-02-01 04:27:45.000000000 +0100
+++ b/sysdeps/unix/sysv/linux/x86_64/64/configure.ac	2023-04-06 21:38:47.647320051 +0200
@@ -1,4 +1,4 @@
 GLIBC_PROVIDES dnl See aclocal.m4 in the top level source directory.
 # Local configure fragment for sysdeps/unix/sysv/linux/x86_64/64.
 
-LIBC_SLIBDIR_RTLDDIR([lib64], [lib64])
+LIBC_SLIBDIR_RTLDDIR([lib], [lib])
diff -Naur a/sysdeps/unix/sysv/linux/x86_64/ldd-rewrite.sed b/sysdeps/unix/sysv/linux/x86_64/ldd-rewrite.sed
--- a/sysdeps/unix/sysv/linux/x86_64/ldd-rewrite.sed	2023-02-01 04:27:45.000000000 +0100
+++ b/sysdeps/unix/sysv/linux/x86_64/ldd-rewrite.sed	2023-04-06 21:41:57.111708090 +0200
@@ -1 +1 @@
-s_^\(RTLDLIST=\)\(.*lib\)\(\|64\|x32\)\(/[^/]*\)\(-x86-64\|-x32\)\(\.so\.[0-9.]*\)[ 	]*$_\1"\2\4\6 \264\4-x86-64\6 \2x32\4-x32\6"_
+s_^\(RTLDLIST=\)\(.*lib\)\(\|64\|x32\)\(/[^/]*\)\(-x86-64\|-x32\)\(\.so\.[0-9.]*\)[ 	]*$_\1"\2\4\6 \2\4-x86-64\6 \2x32\4-x32\6"_
