diff -Naur a/bin/aclocal.in b/bin/aclocal.in
--- a/bin/aclocal.in	2024-07-12 07:04:24.000000000 +0200
+++ b/bin/aclocal.in	2025-05-17 03:31:48.242854257 +0200
@@ -66,7 +66,7 @@
 # with the '--system-acdir' option.
 my @user_includes = ();
 my @automake_includes = ('@datadir@/aclocal-' . $APIVERSION);
-my @system_includes = ('@datadir@/aclocal');
+my @system_includes = ('@datadir@/aclocal', '@datadir@/gettext');
 my $aclocal_path = '';
 
 # Whether we should copy M4 file in $user_includes[0].
