diff -Naur a/subprojects/gssdp-1.2.wrap b/subprojects/gssdp-1.2.wrap
--- a/subprojects/gssdp-1.2.wrap	2023-04-22 20:51:10.000000000 +0200
+++ b/subprojects/gssdp-1.2.wrap	2023-07-25 16:42:50.639646512 +0200
@@ -3,6 +3,6 @@
 revision = master
 depth = 1
  
-[provides]
-dependency_name = gssdp-1.2
+[provide]
+dependency_names = gssdp-1.2
 
diff -Naur a/subprojects/gupnp-1.2.wrap b/subprojects/gupnp-1.2.wrap
--- a/subprojects/gupnp-1.2.wrap	2023-04-22 20:51:10.000000000 +0200
+++ b/subprojects/gupnp-1.2.wrap	2023-07-25 16:42:59.055671783 +0200
@@ -3,6 +3,6 @@
 revision = master
 depth = 1
 
-[provides]
-dependency_name = gupnp-1.2
+[provide]
+dependency_names = gupnp-1.2
 
diff -Naur a/subprojects/gupnp-av-1.0.wrap b/subprojects/gupnp-av-1.0.wrap
--- a/subprojects/gupnp-av-1.0.wrap	2023-04-22 20:51:10.000000000 +0200
+++ b/subprojects/gupnp-av-1.0.wrap	2023-07-25 16:43:19.707729730 +0200
@@ -3,6 +3,6 @@
 revision = master
 depth = 1
  
-[provides]
-dependency_name = gupnp-av-1.0
+[provide]
+dependency_names = gupnp-av-1.0
 
diff -Naur a/subprojects/gupnp-dlna-2.0.wrap b/subprojects/gupnp-dlna-2.0.wrap
--- a/subprojects/gupnp-dlna-2.0.wrap	2023-04-22 20:51:10.000000000 +0200
+++ b/subprojects/gupnp-dlna-2.0.wrap	2023-07-25 16:43:12.699710061 +0200
@@ -3,6 +3,6 @@
 revision = master
 depth = 1
 
-[provides]
-dependency_name = gupnp-dlna-2.0, gupnp-dlna-gst-2.0
+[provide]
+dependency_names = gupnp-dlna-2.0, gupnp-dlna-gst-2.0
 
