diff -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2025-02-19 16:29:33.000000000 +0100
+++ b/CMakeLists.txt	2025-03-04 21:31:38.258945521 +0100
@@ -414,8 +414,8 @@
 	add_compile_options(-fno-omit-frame-pointer)
 endif()
 
-# try to ensure some compiler sanity and hardening options where supported
-foreach (flag -fno-strict-overflow -fno-delete-null-pointer-checks -fhardened)
+# try to ensure some compiler sanity
+foreach (flag -fno-strict-overflow -fno-delete-null-pointer-checks)
 	check_c_compiler_flag(${flag} found)
 	if (found)
 		add_compile_options(${flag})
