diff -Naur a/Makefile b/Makefile
--- a/Makefile	2023-11-04 17:22:25.000000000 +0100
+++ b/Makefile	2023-11-11 18:24:27.226076941 +0100
@@ -15,14 +15,14 @@
 endif
 
 PREFIX?=/usr
-SBINDIR?=/sbin
+SBINDIR?=$(PREFIX)/sbin
 CONF_ETC_DIR?=/etc/iproute2
 CONF_USR_DIR?=$(LIBDIR)/iproute2
 NETNS_RUN_DIR?=/var/run/netns
 NETNS_ETC_DIR?=/etc/netns
 DATADIR?=$(PREFIX)/share
 HDRDIR?=$(PREFIX)/include/iproute2
-DOCDIR?=$(DATADIR)/doc/iproute2
+DOCDIR?=$(DATADIR)/doc/Packages/iproute2
 MANDIR?=$(DATADIR)/man
 ARPDDIR?=/var/lib/arpd
 KERNEL_INCLUDE?=/usr/include
