diff -Naur a/Makefile b/Makefile
--- a/Makefile	2024-01-08 18:38:23.000000000 +0100
+++ b/Makefile	2024-01-11 21:13:51.019236095 +0100
@@ -15,14 +15,14 @@
 endif
 
 PREFIX?=/usr
-SBINDIR?=/sbin
+SBINDIR?=$(PREFIX)/sbin
 NETNS_RUN_DIR?=/var/run/netns
 NETNS_ETC_DIR?=/etc/netns
 DATADIR?=$(PREFIX)/share
 HDRDIR?=$(PREFIX)/include/iproute2
 CONF_ETC_DIR?=/etc/iproute2
 CONF_USR_DIR?=$(DATADIR)/iproute2
-DOCDIR?=$(DATADIR)/doc/iproute2
+DOCDIR?=$(DATADIR)/doc/Packages/iproute2
 MANDIR?=$(DATADIR)/man
 ARPDDIR?=/var/lib/arpd
 KERNEL_INCLUDE?=/usr/include
