diff -Naur a/config.h.meson b/config.h.meson
--- a/config.h.meson	2025-10-12 10:35:02.000000000 +0200
+++ b/config.h.meson	2025-10-17 23:47:27.986574296 +0200
@@ -37,6 +37,3 @@
 
 /* Define if libmtp support is enabled */
 #mesondefine WITH_LIBMTP
-
-/* Define if using girepository 2.0 */
-#mesondefine USE_GIREPOSITORY2
diff -Naur a/meson.build b/meson.build
--- a/meson.build	2025-10-12 10:35:02.000000000 +0200
+++ b/meson.build	2025-10-17 23:47:27.986574296 +0200
@@ -51,7 +51,6 @@
 gio_unix = dependency('gio-unix-2.0', version: '>=' + glib_required, required: true)
 glib = dependency('glib-2.0', version: '>=' + glib_required, required: true)
 gobject = dependency('gobject-2.0', version: '>=' + glib_required, required: true)
-girepository = dependency('girepository-2.0', version: '>=' + glib_required, required: false)
 gobject_introspection = dependency('gobject-introspection-1.0', version: '>= 0.10.0', required: true)
 gstreamer = dependency('gstreamer-1.0', version: '>= ' + gstreamer_required, required: true)
 gstreamer_audio = dependency('gstreamer-audio-1.0', version: '>= ' + gstreamer_required, required: true)
@@ -75,10 +74,9 @@
   cdata.set('HAVE_GUDEV', 1)
 endif
 
-
 common_deps = [
   libmath,
-  intl, glib, gobject, gio, gio_unix,
+  intl, glib, gobject, gio, gio_unix, gobject_introspection,
   cairo, pango, gdk_pixbuf, gtk,
   gstreamer, gstreamer_audio, gstreamer_base, gstreamer_controller, gstreamer_plugins_base, gstreamer_pbutils, gstreamer_tag,
   json_glib, libsoup, libxml, tdb, totem_plparser,
@@ -144,27 +142,12 @@
 sampleplugindatadir = datadir / 'rhythmbox' / 'sample-plugins'
 
 enable_python = false
-enable_gir2 = false
 python_install = python.find_installation('python3', required: get_option('plugins_python'))
 pygobject = dependency('pygobject-3.0', version: '>= 3.0.0', required: get_option('plugins_python'))
 if python_install.found() and pygobject.found()
   enable_python = true
-
-  if pygobject.version() >= '3.53.0'
-    # must have girepository-2.0 and a compatible version of libpeas for this to work
-    if girepository.found() and libpeas.version() > '1.36'
-      enable_gir2 = true
-      cdata.set('USE_GIREPOSITORY2', 1)
-    else
-      error('cannot mix girepository 1.0 (via libpeas) and 2.0 (via pygobject)')
-    endif
-  elif pygobject.version() >= '3.52.0'
-    error('rhythmbox cannot be used with pygobject 3.52 due to girepository clashes')
-  endif
 endif
 
-
-
 enable_vala = false
 vala_found = add_languages('vala', required: get_option('plugins_vala'), native: false)
 if vala_found
@@ -318,7 +301,6 @@
          'libnotify support': plugins.contains('notification'),
          'GUdev support': gudev.found(),
          'Python plugin support': enable_python,
-         'Use girepository 2.0': enable_gir2,
          'Vala plugin support': enable_vala,
          'Libsecret keyring support': libsecret.found(),
          'FM radio support': plugins.contains('fmradio'),
diff -Naur a/shell/main.c b/shell/main.c
--- a/shell/main.c	2025-10-12 10:35:02.000000000 +0200
+++ b/shell/main.c	2025-10-17 23:47:27.986574296 +0200
@@ -39,6 +39,8 @@
 #include <X11/Xlib.h>
 #endif
 
+#include <girepository.h>
+
 #include "rb-shell.h"
 #include "rb-util.h"
 #include "rb-debug.h"
diff -Naur a/shell/meson.build b/shell/meson.build
--- a/shell/meson.build	2025-10-12 10:35:02.000000000 +0200
+++ b/shell/meson.build	2025-10-17 23:47:27.986574296 +0200
@@ -86,25 +86,12 @@
   podcastinc, sourcesinc, widgetsinc, include_directories('.'),
 ]
 
-rhythmbox_core_deps = [
-  common_deps, mediaplayerid_dep, librb_dep, rhythmdb_dep, rbbackends_dep, rbmetadata_dep,
-  rbpodcast_dep, rbwidgets_dep, sources_dep, libmtp
-]
-
-if enable_python
-  rhythmbox_core_deps += [pygobject]
-  if enable_gir2
-    rhythmbox_core_deps += [girepository]
-  else
-    rhythmbox_core_deps += [gobject_introspection]
-  endif
-endif
-
 librhythmbox_core = shared_library('rhythmbox-core',
   shell_sources,
   c_args: shell_c_args,
   include_directories: rhythmbox_core_include_directories,
-  dependencies: rhythmbox_core_deps,
+  dependencies: [common_deps, mediaplayerid_dep, librb_dep, rhythmdb_dep,
+    rbbackends_dep, rbmetadata_dep, rbpodcast_dep, rbwidgets_dep, sources_dep, libmtp],
   install: true,
   link_whole: [librb, rbwidgets_lib, sources_lib],
   version: '@0@.@1@.@2@'.format(RHYTHMBOX_CORE_CURRENT, RHYTHMBOX_CORE_REVISION, RHYTHMBOX_CORE_AGE)
@@ -137,6 +124,11 @@
   install: true,
 )
 
+rhythmbox_dependencies = [rhythmbox_core_dep]
+if enable_python
+  rhythmbox_dependencies += [pygobject]
+endif
+
 # set rpath on the rhythmbox binary if we're installing inside
 # the source dir, ensuring it loads librhythmbox-core from this
 # build rather than a package installed on the system.
@@ -160,7 +152,7 @@
 endif
 
 executable('rhythmbox', 'main.c',
-  dependencies: [ rhythmbox_core_dep ],
+  dependencies: [ rhythmbox_dependencies ],
   c_args: shell_c_args,
   include_directories: [ pluginsinc ],
   install: true,
diff -Naur a/shell/rb-shell.c b/shell/rb-shell.c
--- a/shell/rb-shell.c	2025-10-12 10:35:02.000000000 +0200
+++ b/shell/rb-shell.c	2025-10-17 23:47:27.986574296 +0200
@@ -46,16 +46,7 @@
 #include <glib/gi18n.h>
 #include <gdk/gdk.h>
 #include <gtk/gtk.h>
-
-#ifdef USE_GIREPOSITORY2
-#include <girepository/girepository.h>
-#define g_irepository_require_private	gi_repository_require_private
-#define g_irepository_require		gi_repository_require
-/* who cares about reference counts for a global singleton */
-#define g_irepository_get_default	gi_repository_dup_default
-#else
 #include <girepository.h>
-#endif
 
 #include <libpeas/peas.h>
 #include <libpeas-gtk/peas-gtk.h>
