diff -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2021-04-01 02:17:30.000000000 +0200
+++ b/CMakeLists.txt	2025-07-30 00:45:44.287081409 +0200
@@ -4,7 +4,7 @@
 # http://www.portaudio.com/trac/wiki/TutorialDir/Compile/CMake
 #
 
-CMAKE_MINIMUM_REQUIRED(VERSION 2.8)
+CMAKE_MINIMUM_REQUIRED(VERSION 3.5)
 
 # Check if the user is building PortAudio stand-alone or as part of a larger
 # project. If this is part of a larger project (i.e. the CMakeLists.txt has
