diff -Naur a/Modules/Compiler/Absoft-Fortran.cmake b/Modules/Compiler/Absoft-Fortran.cmake
--- a/Modules/Compiler/Absoft-Fortran.cmake	2024-12-05 14:42:47.000000000 +0100
+++ b/Modules/Compiler/Absoft-Fortran.cmake	2024-12-07 23:41:37.959816356 +0100
@@ -1,7 +1,7 @@
 string(APPEND CMAKE_Fortran_FLAGS_INIT " ")
 string(APPEND CMAKE_Fortran_FLAGS_DEBUG_INIT " -g")
 string(APPEND CMAKE_Fortran_FLAGS_MINSIZEREL_INIT " ")
-string(APPEND CMAKE_Fortran_FLAGS_RELEASE_INIT " -O3")
+string(APPEND CMAKE_Fortran_FLAGS_RELEASE_INIT " -O2")
 string(APPEND CMAKE_Fortran_FLAGS_RELWITHDEBINFO_INIT " -O2 -g")
 set(CMAKE_Fortran_MODDIR_FLAG "-YMOD_OUT_DIR=")
 set(CMAKE_Fortran_MODPATH_FLAG "-p")
diff -Naur a/Modules/Compiler/Fujitsu.cmake b/Modules/Compiler/Fujitsu.cmake
--- a/Modules/Compiler/Fujitsu.cmake	2024-12-05 14:42:47.000000000 +0100
+++ b/Modules/Compiler/Fujitsu.cmake	2024-12-07 23:45:24.752389617 +0100
@@ -16,7 +16,7 @@
   # Initial configuration flags
   string(APPEND CMAKE_${lang}_FLAGS_INIT " ")
   string(APPEND CMAKE_${lang}_FLAGS_DEBUG_INIT " -g -O0")
-  string(APPEND CMAKE_${lang}_FLAGS_RELEASE_INIT " -O3 -DNDEBUG")
+  string(APPEND CMAKE_${lang}_FLAGS_RELEASE_INIT " -O2 -DNDEBUG")
   string(APPEND CMAKE_${lang}_FLAGS_RELWITHDEBINFO_INIT " -O2 -g -DNDEBUG")
 
   # PIC flags
diff -Naur a/Modules/Compiler/G95-Fortran.cmake b/Modules/Compiler/G95-Fortran.cmake
--- a/Modules/Compiler/G95-Fortran.cmake	2024-12-05 14:42:47.000000000 +0100
+++ b/Modules/Compiler/G95-Fortran.cmake	2024-12-07 23:41:46.419837766 +0100
@@ -1,7 +1,7 @@
 string(APPEND CMAKE_Fortran_FLAGS_INIT " ")
 string(APPEND CMAKE_Fortran_FLAGS_DEBUG_INIT " -g")
 string(APPEND CMAKE_Fortran_FLAGS_MINSIZEREL_INIT " -Os")
-string(APPEND CMAKE_Fortran_FLAGS_RELEASE_INIT " -O3")
+string(APPEND CMAKE_Fortran_FLAGS_RELEASE_INIT " -O2")
 string(APPEND CMAKE_Fortran_FLAGS_RELWITHDEBINFO_INIT " -O2 -g")
 set(CMAKE_Fortran_MODDIR_FLAG "-fmod=")
 set(CMAKE_Fortran_VERBOSE_FLAG "-v")
diff -Naur a/Modules/Compiler/GNU.cmake b/Modules/Compiler/GNU.cmake
--- a/Modules/Compiler/GNU.cmake	2024-12-05 14:42:47.000000000 +0100
+++ b/Modules/Compiler/GNU.cmake	2024-12-07 23:42:38.331969086 +0100
@@ -56,7 +56,7 @@
   string(APPEND CMAKE_${lang}_FLAGS_INIT " ")
   string(APPEND CMAKE_${lang}_FLAGS_DEBUG_INIT " -g")
   string(APPEND CMAKE_${lang}_FLAGS_MINSIZEREL_INIT " -Os")
-  string(APPEND CMAKE_${lang}_FLAGS_RELEASE_INIT " -O3")
+  string(APPEND CMAKE_${lang}_FLAGS_RELEASE_INIT " -O2")
   string(APPEND CMAKE_${lang}_FLAGS_RELWITHDEBINFO_INIT " -O2 -g")
   if(NOT "x${lang}" STREQUAL "xFortran")
     string(APPEND CMAKE_${lang}_FLAGS_MINSIZEREL_INIT " -DNDEBUG")
diff -Naur a/Modules/Compiler/Intel.cmake b/Modules/Compiler/Intel.cmake
--- a/Modules/Compiler/Intel.cmake	2024-12-05 14:42:47.000000000 +0100
+++ b/Modules/Compiler/Intel.cmake	2024-12-07 23:43:09.660048299 +0100
@@ -27,7 +27,7 @@
     string(APPEND CMAKE_${lang}_FLAGS_INIT " ")
     string(APPEND CMAKE_${lang}_FLAGS_DEBUG_INIT " -g")
     string(APPEND CMAKE_${lang}_FLAGS_MINSIZEREL_INIT " -Os")
-    string(APPEND CMAKE_${lang}_FLAGS_RELEASE_INIT " -O3")
+    string(APPEND CMAKE_${lang}_FLAGS_RELEASE_INIT " -O2")
     string(APPEND CMAKE_${lang}_FLAGS_RELWITHDEBINFO_INIT " -O2 -g")
 
     # Compiler + IPO does not recognize --dependency-file link option
diff -Naur a/Modules/Compiler/Intel-ISPC.cmake b/Modules/Compiler/Intel-ISPC.cmake
--- a/Modules/Compiler/Intel-ISPC.cmake	2024-12-05 14:42:47.000000000 +0100
+++ b/Modules/Compiler/Intel-ISPC.cmake	2024-12-07 23:46:00.172479050 +0100
@@ -13,7 +13,7 @@
 
 string(APPEND CMAKE_ISPC_FLAGS_INIT " ")
 string(APPEND CMAKE_ISPC_FLAGS_DEBUG_INIT "-O0 -g")
-string(APPEND CMAKE_ISPC_FLAGS_RELEASE_INIT " -O3 -DNDEBUG")
+string(APPEND CMAKE_ISPC_FLAGS_RELEASE_INIT " -O2 -DNDEBUG")
 string(APPEND CMAKE_ISPC_FLAGS_MINSIZEREL_INIT " -O1 -DNDEBUG")
 string(APPEND CMAKE_ISPC_FLAGS_RELWITHDEBINFO_INIT " -O2 -g -DNDEBUG")
 
diff -Naur a/Modules/Compiler/IntelLLVM.cmake b/Modules/Compiler/IntelLLVM.cmake
--- a/Modules/Compiler/IntelLLVM.cmake	2024-12-05 14:42:47.000000000 +0100
+++ b/Modules/Compiler/IntelLLVM.cmake	2024-12-07 23:45:49.648452480 +0100
@@ -58,7 +58,7 @@
       endif()
     endif()
     string(APPEND CMAKE_${lang}_FLAGS_MINSIZEREL_INIT " -Os")
-    string(APPEND CMAKE_${lang}_FLAGS_RELEASE_INIT " -O3")
+    string(APPEND CMAKE_${lang}_FLAGS_RELEASE_INIT " -O2")
     string(APPEND CMAKE_${lang}_FLAGS_RELWITHDEBINFO_INIT " -O2 -g")
 
     set(_CMAKE_${lang}_PIE_MAY_BE_SUPPORTED_BY_LINKER YES)
diff -Naur a/Modules/Compiler/LCC.cmake b/Modules/Compiler/LCC.cmake
--- a/Modules/Compiler/LCC.cmake	2024-12-05 14:42:47.000000000 +0100
+++ b/Modules/Compiler/LCC.cmake	2024-12-07 23:44:47.116294569 +0100
@@ -52,7 +52,7 @@
   string(APPEND CMAKE_${lang}_FLAGS_INIT " ")
   string(APPEND CMAKE_${lang}_FLAGS_DEBUG_INIT " -g")
   string(APPEND CMAKE_${lang}_FLAGS_MINSIZEREL_INIT " -Os -DNDEBUG")
-  string(APPEND CMAKE_${lang}_FLAGS_RELEASE_INIT " -O3 -DNDEBUG")
+  string(APPEND CMAKE_${lang}_FLAGS_RELEASE_INIT " -O2 -DNDEBUG")
   string(APPEND CMAKE_${lang}_FLAGS_RELWITHDEBINFO_INIT " -O2 -g -DNDEBUG")
   set(CMAKE_${lang}_CREATE_PREPROCESSED_SOURCE "<CMAKE_${lang}_COMPILER> <DEFINES> <INCLUDES> <FLAGS> -E <SOURCE> > <PREPROCESSED_SOURCE>")
   set(CMAKE_${lang}_CREATE_ASSEMBLY_SOURCE "<CMAKE_${lang}_COMPILER> <DEFINES> <INCLUDES> <FLAGS> -S <SOURCE> -o <ASSEMBLY_SOURCE>")
diff -Naur a/Modules/Compiler/LCC-Fortran.cmake b/Modules/Compiler/LCC-Fortran.cmake
--- a/Modules/Compiler/LCC-Fortran.cmake	2024-12-05 14:42:47.000000000 +0100
+++ b/Modules/Compiler/LCC-Fortran.cmake	2024-12-07 23:44:57.812321588 +0100
@@ -20,7 +20,7 @@
 
 # No -DNDEBUG for Fortran.
 string(APPEND CMAKE_Fortran_FLAGS_MINSIZEREL_INIT " -Os")
-string(APPEND CMAKE_Fortran_FLAGS_RELEASE_INIT " -O3")
+string(APPEND CMAKE_Fortran_FLAGS_RELEASE_INIT " -O2")
 
 # No -isystem for Fortran because it will not find .mod files.
 unset(CMAKE_INCLUDE_SYSTEM_FLAG_Fortran)
diff -Naur a/Modules/Compiler/LFortran-Fortran.cmake b/Modules/Compiler/LFortran-Fortran.cmake
--- a/Modules/Compiler/LFortran-Fortran.cmake	2024-12-05 14:42:47.000000000 +0100
+++ b/Modules/Compiler/LFortran-Fortran.cmake	2024-12-07 23:45:07.980347268 +0100
@@ -1,6 +1,6 @@
 string(APPEND CMAKE_Fortran_FLAGS_DEBUG_INIT " -g")
 string(APPEND CMAKE_Fortran_FLAGS_MINSIZEREL_INIT " ")
-string(APPEND CMAKE_Fortran_FLAGS_RELEASE_INIT " -O3")
+string(APPEND CMAKE_Fortran_FLAGS_RELEASE_INIT " -O2")
 string(APPEND CMAKE_Fortran_FLAGS_RELWITHDEBINFO_INIT " -O2 -g")
 set(CMAKE_Fortran_MODDIR_FLAG "-J")
 set(CMAKE_Fortran_VERBOSE_FLAG "-v -Wl,-v")
diff -Naur a/Modules/Compiler/LLVMFlang-Fortran.cmake b/Modules/Compiler/LLVMFlang-Fortran.cmake
--- a/Modules/Compiler/LLVMFlang-Fortran.cmake	2024-12-05 14:42:47.000000000 +0100
+++ b/Modules/Compiler/LLVMFlang-Fortran.cmake	2024-12-07 23:45:35.052415629 +0100
@@ -23,5 +23,5 @@
 
   string(APPEND CMAKE_Fortran_FLAGS_DEBUG_INIT " -O0 -g")
   string(APPEND CMAKE_Fortran_FLAGS_RELWITHDEBINFO_INIT " -O2 -g")
-  string(APPEND CMAKE_Fortran_FLAGS_RELEASE_INIT " -O3")
+  string(APPEND CMAKE_Fortran_FLAGS_RELEASE_INIT " -O2")
 endif()
diff -Naur a/Modules/Compiler/NVIDIA.cmake b/Modules/Compiler/NVIDIA.cmake
--- a/Modules/Compiler/NVIDIA.cmake	2024-12-05 14:42:47.000000000 +0100
+++ b/Modules/Compiler/NVIDIA.cmake	2024-12-07 23:43:38.020119986 +0100
@@ -139,7 +139,7 @@
     set(CMAKE_SHARED_LIBRARY_${lang}_FLAGS -fPIC)
     string(APPEND CMAKE_${lang}_FLAGS_INIT " ")
     string(APPEND CMAKE_${lang}_FLAGS_DEBUG_INIT " -g")
-    string(APPEND CMAKE_${lang}_FLAGS_RELEASE_INIT " -O3 -DNDEBUG")
+    string(APPEND CMAKE_${lang}_FLAGS_RELEASE_INIT " -O2 -DNDEBUG")
     string(APPEND CMAKE_${lang}_FLAGS_MINSIZEREL_INIT " -O1 -DNDEBUG")
     string(APPEND CMAKE_${lang}_FLAGS_RELWITHDEBINFO_INIT " -O2 -g -DNDEBUG")
   endif()
diff -Naur a/Modules/Compiler/PathScale.cmake b/Modules/Compiler/PathScale.cmake
--- a/Modules/Compiler/PathScale.cmake	2024-12-05 14:42:47.000000000 +0100
+++ b/Modules/Compiler/PathScale.cmake	2024-12-07 23:43:51.428153868 +0100
@@ -16,6 +16,6 @@
   string(APPEND CMAKE_${lang}_FLAGS_INIT " ")
   string(APPEND CMAKE_${lang}_FLAGS_DEBUG_INIT " -g -O0")
   string(APPEND CMAKE_${lang}_FLAGS_MINSIZEREL_INIT " -Os")
-  string(APPEND CMAKE_${lang}_FLAGS_RELEASE_INIT " -O3")
+  string(APPEND CMAKE_${lang}_FLAGS_RELEASE_INIT " -O2")
   string(APPEND CMAKE_${lang}_FLAGS_RELWITHDEBINFO_INIT " -g -O2")
 endmacro()
diff -Naur a/Modules/Compiler/PGI.cmake b/Modules/Compiler/PGI.cmake
--- a/Modules/Compiler/PGI.cmake	2024-12-05 14:42:47.000000000 +0100
+++ b/Modules/Compiler/PGI.cmake	2024-12-07 23:44:16.360216865 +0100
@@ -18,7 +18,7 @@
   string(APPEND CMAKE_${lang}_FLAGS_INIT " ")
   string(APPEND CMAKE_${lang}_FLAGS_DEBUG_INIT " -g -O0")
   string(APPEND CMAKE_${lang}_FLAGS_MINSIZEREL_INIT " -O2 -s")
-  string(APPEND CMAKE_${lang}_FLAGS_RELEASE_INIT " -fast -O3")
+  string(APPEND CMAKE_${lang}_FLAGS_RELEASE_INIT " -O2")
   string(APPEND CMAKE_${lang}_FLAGS_RELWITHDEBINFO_INIT " -O2 -gopt")
 
   if(CMAKE_HOST_WIN32)
