diff -Naur a/meson.build b/meson.build
--- a/meson.build	2023-09-22 16:17:10.000000000 +0200
+++ b/meson.build	2023-10-02 08:48:10.369126705 +0200
@@ -7,9 +7,9 @@
   gdk_pb_moddir = gdkpb.get_variable(pkgconfig: 'gdk_pixbuf_moduledir', pkgconfig_define: ['prefix', get_option('prefix')])
 endif
 
-webp = dependency('libwebp', version: '>1.3.2')
-webpdemux = dependency('libwebpdemux', version: '>1.3.2')
-webpmux = dependency('libwebpmux', version: '>1.3.2')
+webp = dependency('libwebp', version: '>=1.3.2')
+webpdemux = dependency('libwebpdemux', version: '>=1.3.2')
+webpmux = dependency('libwebpmux', version: '>=1.3.2')
 
 pbl_webp = shared_module('pixbufloader-webp',
                          sources: ['io-webp.c', 'io-webp-anim.c', 'io-webp-anim-iter.c'],
