diff -Naur a/src/aafire.c b/src/aafire.c
--- a/src/aafire.c	2001-04-26 17:30:03.000000000 +0200
+++ b/src/aafire.c	2024-06-30 11:53:47.626991998 +0200
@@ -1,3 +1,4 @@
+#include <stdlib.h>
 #include <stdio.h>
 #include "aalib.h"
 
diff -Naur a/src/aainfo.c b/src/aainfo.c
--- a/src/aainfo.c	2001-04-26 16:37:31.000000000 +0200
+++ b/src/aainfo.c	2024-06-30 11:53:47.626991998 +0200
@@ -1,3 +1,4 @@
+#include <stdlib.h>
 
 #include "aalib.h"
 #include "aaint.h"
diff -Naur a/src/aakbdreg.c b/src/aakbdreg.c
--- a/src/aakbdreg.c	2001-04-26 16:37:31.000000000 +0200
+++ b/src/aakbdreg.c	2024-06-30 11:53:47.630992018 +0200
@@ -1,3 +1,4 @@
+#include <string.h>
 #include <malloc.h>
 #include "config.h"
 #include "aalib.h"
diff -Naur a/src/aalib.c b/src/aalib.c
--- a/src/aalib.c	2001-04-26 16:37:31.000000000 +0200
+++ b/src/aalib.c	2024-06-30 11:53:47.626991998 +0200
@@ -1,3 +1,4 @@
+#include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
 #include <malloc.h>
diff -Naur a/src/aalinuxkbd.c b/src/aalinuxkbd.c
--- a/src/aalinuxkbd.c	2001-04-26 16:37:31.000000000 +0200
+++ b/src/aalinuxkbd.c	2024-06-30 11:53:47.638992058 +0200
@@ -1,3 +1,4 @@
+#include <stdlib.h>
 /* Raw keyboard mode driver for linux console by Jan Hubicka.
  * This driver is based on:
  * librawkey v0.21 - (c) 1994, 1995 Russell Marks
@@ -200,7 +201,7 @@
 static int rawmode_init(void)
 {
     if (!closed)
-	return;
+	return 0;
     mypid = getpid();
     if (tty_fd == -1) {
 	tty_fd = fileno(stdin);
diff -Naur a/src/aamoureg.c b/src/aamoureg.c
--- a/src/aamoureg.c	2001-04-26 16:37:31.000000000 +0200
+++ b/src/aamoureg.c	2024-06-30 11:53:47.630992018 +0200
@@ -1,3 +1,4 @@
+#include <string.h>
 #include <malloc.h>
 #include "config.h"
 #include "aalib.h"
diff -Naur a/src/aaregist.c b/src/aaregist.c
--- a/src/aaregist.c	2001-04-26 16:37:31.000000000 +0200
+++ b/src/aaregist.c	2024-06-30 11:53:47.634992038 +0200
@@ -1,3 +1,5 @@
+#include <string.h>
+#include <stdlib.h>
 #include <malloc.h>
 #include "config.h"
 #include "aalib.h"
diff -Naur a/src/aasavefont.c b/src/aasavefont.c
--- a/src/aasavefont.c	2001-04-26 16:37:31.000000000 +0200
+++ b/src/aasavefont.c	2024-06-30 11:53:47.630992018 +0200
@@ -1,3 +1,4 @@
+#include <stdlib.h>
 #include "aalib.h"
 int main(int argc, char **argv)
 {
diff -Naur a/src/aatest.c b/src/aatest.c
--- a/src/aatest.c	2001-04-26 16:37:31.000000000 +0200
+++ b/src/aatest.c	2024-06-30 11:53:47.630992018 +0200
@@ -1,3 +1,5 @@
+#include <string.h>
+#include <stdlib.h>
 #include "aalib.h"
 int main(int argc, char **argv)
 {
diff -Naur a/src/aaxkbd.c b/src/aaxkbd.c
--- a/src/aaxkbd.c	2001-04-26 16:37:31.000000000 +0200
+++ b/src/aaxkbd.c	2024-06-30 11:53:47.638992058 +0200
@@ -4,6 +4,7 @@
 #include <stdio.h>
 #include "config.h"
 #ifdef X11_KBDDRIVER
+#include <X11/Xutil.h>
 #include <X11/Xlib.h>
 #include <X11/keysymdef.h>
 #include <X11/keysym.h>
