diff -Naur a/usbutils.pc.in b/usbutils.pc.in
--- a/usbutils.pc.in	2023-10-22 13:32:14.000000000 +0200
+++ b/usbutils.pc.in	2023-10-26 18:29:21.927196982 +0200
@@ -1,7 +1,7 @@
 prefix=@prefix@
 exec_prefix=@prefix@
-libdir=@prefix@/lib64
-includedir=@prefix@/include
+libdir=@libdir@
+includedir=@includedir@
 usbids=@datadir@/usb.ids
 
 Name: @PACKAGE_NAME@
@@ -13,6 +13,6 @@
 URL: https://github.com/gregkh/usbutils
 Requires: libusb-1.0 >= 1.0.14  libudev >= 196
 Conflicts:
-Libs: -L@prefix@/lib64
+Libs: -L${libdir}
 Libs.private: @LIBUSB_LIBS@ @UDEV_LIBS@
 Cflags: @CFLAGS@ @LIBUSB_CFLAGS@ @UDEV_CFLAGS@
