diff -Naur a/drivers/configure b/drivers/configure
--- a/drivers/configure	2013-03-12 00:52:13.000000000 +0100
+++ b/drivers/configure	2025-04-20 22:54:09.418423940 +0200
@@ -13574,7 +13574,7 @@
 done
 
 
-ac_config_files="$ac_config_files libdbi-drivers.spec Makefile doc/Makefile doc/include/Makefile drivers/Makefile drivers/mysql/Makefile drivers/pgsql/Makefile drivers/sqlite/Makefile drivers/sqlite3/Makefile drivers/msql/Makefile drivers/oracle/Makefile drivers/firebird/Makefile drivers/freetds/Makefile drivers/ingres/Makefile drivers/db2/Makefile tests/Makefile tests/cgreen/Makefile tests/test_dbi.cfg tests/plugin_settings.sh"
+ac_config_files="$ac_config_files libdbi-drivers.spec Makefile doc/Makefile doc/include/Makefile drivers/Makefile drivers/mysql/Makefile drivers/pgsql/Makefile drivers/sqlite/Makefile drivers/sqlite3/Makefile drivers/msql/Makefile drivers/oracle/Makefile drivers/firebird/Makefile drivers/freetds/Makefile drivers/ingres/Makefile drivers/db2/Makefile"
 
 cat >confcache <<\_ACEOF
 # This file is a shell script that caches the results of configure
@@ -14690,10 +14690,6 @@
     "drivers/freetds/Makefile") CONFIG_FILES="$CONFIG_FILES drivers/freetds/Makefile" ;;
     "drivers/ingres/Makefile") CONFIG_FILES="$CONFIG_FILES drivers/ingres/Makefile" ;;
     "drivers/db2/Makefile") CONFIG_FILES="$CONFIG_FILES drivers/db2/Makefile" ;;
-    "tests/Makefile") CONFIG_FILES="$CONFIG_FILES tests/Makefile" ;;
-    "tests/cgreen/Makefile") CONFIG_FILES="$CONFIG_FILES tests/cgreen/Makefile" ;;
-    "tests/test_dbi.cfg") CONFIG_FILES="$CONFIG_FILES tests/test_dbi.cfg" ;;
-    "tests/plugin_settings.sh") CONFIG_FILES="$CONFIG_FILES tests/plugin_settings.sh" ;;
 
   *) as_fn_error $? "invalid argument: \`$ac_config_target'" "$LINENO" 5;;
   esac
diff -Naur a/drivers/configure.in b/drivers/configure.in
--- a/drivers/configure.in	2013-03-12 00:41:05.000000000 +0100
+++ b/drivers/configure.in	2025-04-20 22:52:58.510271325 +0200
@@ -269,10 +269,6 @@
 	drivers/freetds/Makefile
 	drivers/ingres/Makefile
 	drivers/db2/Makefile
-	tests/Makefile
-	tests/cgreen/Makefile
-	tests/test_dbi.cfg
-    tests/plugin_settings.sh
 	])
 dnl doc/Makefile
 
diff -Naur a/drivers/Makefile.am b/drivers/Makefile.am
--- a/drivers/Makefile.am	2011-05-01 12:42:05.000000000 +0200
+++ b/drivers/Makefile.am	2025-04-20 22:51:56.294137156 +0200
@@ -4,7 +4,7 @@
 
 ACLOCAL_AMFLAGS = -I m4
 
-SUBDIRS = drivers tests @docs_subdirs@
+SUBDIRS = drivers @docs_subdirs@
 
 EXTRA_DIST = AUTHORS COPYING ChangeLog INSTALL README README.win32 README.osx README.encodings TODO libdbi-drivers.spec.in config.h.in Makefile.w32 autogen.sh fixlt.sh
 
diff -Naur a/drivers/Makefile.in b/drivers/Makefile.in
--- a/drivers/Makefile.in	2013-03-12 00:52:10.000000000 +0100
+++ b/drivers/Makefile.in	2025-04-20 22:52:20.374189056 +0200
@@ -303,7 +303,7 @@
 top_srcdir = @top_srcdir@
 AUTOMAKE_OPTIONS = foreign dist-zip
 ACLOCAL_AMFLAGS = -I m4
-SUBDIRS = drivers tests @docs_subdirs@
+SUBDIRS = drivers @docs_subdirs@
 EXTRA_DIST = AUTHORS COPYING ChangeLog INSTALL README README.win32 README.osx README.encodings TODO libdbi-drivers.spec.in config.h.in Makefile.w32 autogen.sh fixlt.sh
 all: config.h
 	$(MAKE) $(AM_MAKEFLAGS) all-recursive
