diff -Naur a/schemas/org.gnome.desktop.background.gschema.xml.in b/schemas/org.gnome.desktop.background.gschema.xml.in
--- a/schemas/org.gnome.desktop.background.gschema.xml.in	2025-09-14 12:59:21.000000000 +0200
+++ b/schemas/org.gnome.desktop.background.gschema.xml.in	2025-09-15 20:40:59.805544216 +0200
@@ -11,7 +11,7 @@
       </description>
     </key>
     <key name="picture-uri" type="s">
-      <default>'file://@datadir@/backgrounds/gnome/adwaita-l.jxl'</default>
+      <default>'file://@datadir@/backgrounds/UHU/uhu_linux_ubk.jpg'</default>
       <summary>Picture URI</summary>
       <description>
         URI to use for the background image. Note that the backend only supports
@@ -19,7 +19,7 @@
       </description>
     </key>
     <key name="picture-uri-dark" type="s">
-      <default>'file://@datadir@/backgrounds/gnome/adwaita-d.jxl'</default>
+      <default>'file://@datadir@/backgrounds/UHU/uhu_linux_ubk.jpg'</default>
       <summary>Picture URI (dark)</summary>
       <description>
         URI to use for the background image. Note that the backend only supports
@@ -57,7 +57,7 @@
       </description>
     </key>
     <key name="show-desktop-icons" type="b">
-      <default>false</default>
+      <default>true</default>
       <summary>Have file manager handle the desktop</summary>
       <description>
         If set to true, then file manager will draw the icons on the desktop.
diff -Naur a/schemas/org.gnome.desktop.input-sources.gschema.xml.in b/schemas/org.gnome.desktop.input-sources.gschema.xml.in
--- a/schemas/org.gnome.desktop.input-sources.gschema.xml.in	2025-09-14 12:59:21.000000000 +0200
+++ b/schemas/org.gnome.desktop.input-sources.gschema.xml.in	2025-09-15 20:41:58.141655116 +0200
@@ -15,7 +15,7 @@
       </description>
     </key>
     <key name="sources" type="a(ss)">
-      <default>[]</default>
+      <default>[('xkb', 'hu'), ('xkb', 'us')]</default>
       <summary>List of input sources</summary>
       <description>
         List of input source identifiers available. Each source is
diff -Naur a/schemas/org.gnome.desktop.screensaver.gschema.xml.in b/schemas/org.gnome.desktop.screensaver.gschema.xml.in
--- a/schemas/org.gnome.desktop.screensaver.gschema.xml.in	2025-09-14 12:59:21.000000000 +0200
+++ b/schemas/org.gnome.desktop.screensaver.gschema.xml.in	2025-09-15 20:42:41.581742570 +0200
@@ -102,7 +102,7 @@
       </description>
     </key>
     <key name="picture-uri" type="s">
-      <default>'file://@datadir@/backgrounds/gnome/adwaita-timed.xml'</default>
+      <default>'file://@datadir@/backgrounds/UHU/uhu_linux_ubk.jpg'</default>
       <summary>Picture URI</summary>
       <description>
         URI to use for the background image. Note that the backend only supports
