diff -Naur a/schemas/org.gnome.desktop.input-sources.gschema.xml.in b/schemas/org.gnome.desktop.input-sources.gschema.xml.in
--- a/schemas/org.gnome.desktop.input-sources.gschema.xml.in	2023-09-16 18:49:23.000000000 +0200
+++ b/schemas/org.gnome.desktop.input-sources.gschema.xml.in	2023-09-18 12:07:23.828996241 +0200
@@ -15,7 +15,7 @@
       </description>
     </key>
     <key name="sources" type="a(ss)">
-      <default>[]</default>
+      <default>[('xkb', 'hu'), ('xkb', 'us')]</default>
       <summary>List of input sources</summary>
       <description>
         List of input source identifiers available. Each source is
