diff -Naur a/schemas/org.gnome.desktop.background.gschema.xml.in b/schemas/org.gnome.desktop.background.gschema.xml.in
--- a/schemas/org.gnome.desktop.background.gschema.xml.in	2023-09-16 18:49:23.000000000 +0200
+++ b/schemas/org.gnome.desktop.background.gschema.xml.in	2023-09-18 12:05:05.788446970 +0200
@@ -11,7 +11,7 @@
       </description>
     </key>
     <key name="picture-uri" type="s">
-      <default>'file://@datadir@/backgrounds/gnome/adwaita-l.jpg'</default>
+      <default>'file://@datadir@/backgrounds/UHU/uhu_linux_ubk.jpg'</default>
       <summary>Picture URI</summary>
       <description>
         URI to use for the background image. Note that the backend only supports
@@ -19,7 +19,7 @@
       </description>
     </key>
     <key name="picture-uri-dark" type="s">
-      <default>'file://@datadir@/backgrounds/gnome/adwaita-d.jpg'</default>
+      <default>'file://@datadir@/backgrounds/UHU/uhu_linux_ubk.jpg'</default>
       <summary>Picture URI (dark)</summary>
       <description>
         URI to use for the background image. Note that the backend only supports
diff -Naur a/schemas/org.gnome.desktop.screensaver.gschema.xml.in b/schemas/org.gnome.desktop.screensaver.gschema.xml.in
--- a/schemas/org.gnome.desktop.screensaver.gschema.xml.in	2023-09-16 18:49:23.000000000 +0200
+++ b/schemas/org.gnome.desktop.screensaver.gschema.xml.in	2023-09-18 12:04:40.236345245 +0200
@@ -95,7 +95,7 @@
       </description>
     </key>
     <key name="picture-uri" type="s">
-      <default>'file://@datadir@/backgrounds/gnome/adwaita-timed.xml'</default>
+      <default>'file://@datadir@/backgrounds/UHU/uhu_linux_ubk.jpg'</default>
       <summary>Picture URI</summary>
       <description>
         URI to use for the background image. Note that the backend only supports
