diff -Naur a/schemas/org.gnome.desktop.background.gschema.xml.in b/schemas/org.gnome.desktop.background.gschema.xml.in
--- a/schemas/org.gnome.desktop.background.gschema.xml.in	2023-09-16 18:49:23.000000000 +0200
+++ b/schemas/org.gnome.desktop.background.gschema.xml.in	2023-09-18 12:08:20.293220780 +0200
@@ -57,7 +57,7 @@
       </description>
     </key>
     <key name="show-desktop-icons" type="b">
-      <default>false</default>
+      <default>true</default>
       <summary>Have file manager handle the desktop</summary>
       <description>
         If set to true, then file manager will draw the icons on the desktop.
