diff -Naur a/configure.ac b/configure.ac
--- a/configure.ac	2023-10-30 08:41:48.000000000 +0100
+++ b/configure.ac	2023-11-29 01:17:46.069765923 +0100
@@ -912,7 +912,7 @@
 dnl
 dnl Domain name i18n support via GNU libidn
 dnl
-PKG_CHECK_MODULES([IDN], [libidn], [
+PKG_CHECK_MODULES([IDN], [libidn2], [
   have_libidn="yes"
   AC_DEFINE([HAVE_IDN], 1, [Define to 1 if you have GNU libidn.])
 ], [
diff -Naur a/src/text/url.c b/src/text/url.c
--- a/src/text/url.c	2023-10-30 08:41:48.000000000 +0100
+++ b/src/text/url.c	2023-11-29 23:31:24.480314956 +0100
@@ -873,7 +873,7 @@
 }
 
 #if defined (HAVE_IDN)
-# include <idna.h>
+# include <idn2.h>
 #elif defined (_WIN32)
 # include <windows.h>
 # include <vlc_charset.h>
@@ -920,9 +920,6 @@
         case IDNA_MALLOC_ERROR:
             errno = ENOMEM;
             return NULL;
-        case IDNA_DLOPEN_ERROR:
-            errno = ENOSYS;
-            return NULL;
         default:
             errno = EINVAL;
             return NULL;
