diff -Naur a/gettext-tools/doc/gettext.texi b/gettext-tools/doc/gettext.texi
--- a/gettext-tools/doc/gettext.texi	2025-05-07 18:18:48.000000000 +0200
+++ b/gettext-tools/doc/gettext.texi	2025-02-11 11:39:58.000000000 +0100
@@ -8935,7 +8895,7 @@
 @file{gettext.m4}, @file{host-cpu-c-abi.m4}, @file{intlmacosx.m4},
 @file{iconv.m4}, @file{lib-ld.m4}, @file{lib-link.m4}, @file{lib-prefix.m4},
 @file{nls.m4}, @file{po.m4}, @file{progtest.m4} from GNU @code{gettext}'s
-@file{@var{prefix}/share/gettext/m4/} directory into a single file.
+@file{m4/} directory into a single file.
 
 If you already have an @file{aclocal.m4} file, then you will have
 to merge the said macro files into your @file{aclocal.m4}.  Note that if
@@ -8950,8 +8910,7 @@
 to copy the files @file{build-to-host.m4}, @file{gettext.m4},
 @file{host-cpu-c-abi.m4}, @file{intlmacosx.m4}, @file{iconv.m4},
 @file{lib-ld.m4}, @file{lib-link.m4}, @file{lib-prefix.m4}, @file{nls.m4},
-@file{po.m4}, @file{progtest.m4} from GNU @code{gettext}'s
-@file{@var{prefix}/share/gettext/m4/}
+@file{po.m4}, @file{progtest.m4} from GNU @code{gettext}'s @file{m4/}
 directory to a subdirectory named @file{m4/} and add the line
 
 @example
@@ -8961,26 +8920,23 @@
 @noindent
 to your top level @file{Makefile.am}.
 
-If you are using GNU @code{automake} 1.12 or newer, it is even easier:
+If you are using GNU @code{automake} 1.10 or newer, it is even easier:
 Add the line
 
 @example
-ACLOCAL_AMFLAGS = -I m4
+ACLOCAL_AMFLAGS = --install -I m4
 @end example
 
 @noindent
-to your top level @file{Makefile.am}, and run
-@samp{aclocal --install --system-acdir=@var{prefix}/share/gettext/m4 -I m4}.
+to your top level @file{Makefile.am}, and run @samp{aclocal --install -I m4}.
 This will copy the needed files to the @file{m4/} subdirectory automatically,
 before updating @file{aclocal.m4}.
 
-Note: This @code{--system-acdir} option should only be used here, once.
-If you were to use it after @code{autopoint} has been run,
-it would destroy the consistency that @code{autopoint} guarantees
-and lead to all sorts of malfunction at build time.
-
 These macros check for the internationalization support functions
-and related information.
+and related informations.  Hopefully, once stabilized, these macros
+might be integrated in the standard Autoconf set, because this
+piece of @code{m4} code will be the same for all projects using GNU
+@code{gettext}.
 
 @node config.h.in
 @subsection @file{config.h.in} at top level
diff -Naur a/gettext-tools/m4/Makefile.am b/gettext-tools/m4/Makefile.am
--- a/gettext-tools/m4/Makefile.am	2025-04-25 10:15:08.000000000 +0200
+++ b/gettext-tools/m4/Makefile.am	2023-10-06 16:25:03.000000000 +0200
@@ -17,14 +17,9 @@
 ## Process this file with automake to produce Makefile.in.
 
 # Files installed for use by gettextize.
-# Note: We do *not* install them into $(datadir)/share/aclocal/ any more,
-# because there, 'aclocal --install' (possibly invoked from 'autoreconf')
-# would see the files and copy them into the user's package, destroying
-# the consistency between po.m4 and po/Makefile.in.in.
-
-pkgdatadir = $(datadir)/gettext
-macrosdir = $(pkgdatadir)/m4
-macros_DATA = \
+
+aclocaldir = @aclocaldir@
+aclocal_DATA = \
   ../../gettext-runtime/gnulib-m4/host-cpu-c-abi.m4 \
   ../../gettext-runtime/gnulib-m4/iconv.m4 \
   ../../gettext-runtime/gnulib-m4/intlmacosx.m4 \
@@ -55,55 +47,3 @@
   exported.m4 \
   hostname.m4 \
   locale-de.m4
-
-
-# List of all file names that we ever stored in $(aclocaldir).
-OLD_M4_FILES = \
-  build-to-host.m4 \
-  codeset.m4 \
-  extern-inline.m4 \
-  fcntl-o.m4 \
-  gettext.m4 \
-  glibc2.m4 \
-  glibc21.m4 \
-  host-cpu-c-abi.m4 \
-  iconv.m4 \
-  intdiv0.m4 \
-  intl.m4 \
-  intldir.m4 \
-  intlmacosx.m4 \
-  intmax.m4 \
-  inttypes.m4 \
-  inttypes-h.m4 \
-  inttypes-pri.m4 \
-  inttypes_h.m4 \
-  isc-posix.m4 \
-  lcmessage.m4 \
-  lib-ld.m4 \
-  lib-link.m4 \
-  lib-prefix.m4 \
-  lock.m4 \
-  longdouble.m4 \
-  longlong.m4 \
-  nls.m4 \
-  po.m4 \
-  printf-posix.m4 \
-  progtest.m4 \
-  signed.m4 \
-  size_max.m4 \
-  stdint_h.m4 \
-  threadlib.m4 \
-  uintmax_t.m4 \
-  ulonglong.m4 \
-  visibility.m4 \
-  wchar_t.m4 \
-  wint_t.m4 \
-  xsize.m4
-
-# We don't install the *.m4 files in $(aclocaldir) any more.
-install-data-local: uninstall-macros
-uninstall-local: uninstall-macros
-uninstall-macros:
-	for f in $(OLD_M4_FILES); do \
-	  rm -f '$(DESTDIR)@aclocaldir@'/$$f; \
-	done
diff -Naur a/gettext-tools/m4/Makefile.in b/gettext-tools/m4/Makefile.in
--- a/gettext-tools/m4/Makefile.in	2025-05-07 18:20:13.000000000 +0200
+++ b/gettext-tools/m4/Makefile.in	2025-02-25 11:25:29.000000000 +0100
@@ -15,10 +15,6 @@
 @SET_MAKE@
 
 # Files installed for use by gettextize.
-# Note: We do *not* install them into $(datadir)/share/aclocal/ any more,
-# because there, 'aclocal --install' (possibly invoked from 'autoreconf')
-# would see the files and copy them into the user's package, destroying
-# the consistency between po.m4 and po/Makefile.in.in.
 
 VPATH = @srcdir@
 am__is_gnu_make = { \
@@ -544,14 +533,14 @@
   || { echo " ( cd '$$dir' && rm -f" $$files ")"; \
        $(am__cd) "$$dir" && echo $$files | $(am__xargs_n) 40 $(am__rm_f); }; \
   }
-am__installdirs = "$(DESTDIR)$(examplesbuildauxdir)" \
-	"$(DESTDIR)$(macrosdir)"
-DATA = $(examplesbuildaux_DATA) $(macros_DATA)
+am__installdirs = "$(DESTDIR)$(aclocaldir)" \
+	"$(DESTDIR)$(examplesbuildauxdir)"
+DATA = $(aclocal_DATA) $(examplesbuildaux_DATA)
 am__tagged_files = $(HEADERS) $(SOURCES) $(TAGS_FILES) $(LISP)
 am__DIST_COMMON = $(srcdir)/Makefile.in \
 	$(top_srcdir)/../build-aux/mkinstalldirs README
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
-pkgdatadir = $(datadir)/gettext
+pkgdatadir = @pkgdatadir@
 pkgincludedir = @pkgincludedir@
 pkglibdir = @pkglibdir@
 pkglibexecdir = @pkglibexecdir@
@@ -3760,8 +3681,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-macrosdir = $(pkgdatadir)/m4
-macros_DATA = \
+aclocal_DATA = \
   ../../gettext-runtime/gnulib-m4/host-cpu-c-abi.m4 \
   ../../gettext-runtime/gnulib-m4/iconv.m4 \
   ../../gettext-runtime/gnulib-m4/intlmacosx.m4 \
@@ -3794,50 +3711,6 @@
   hostname.m4 \
   locale-de.m4
 
-
-# List of all file names that we ever stored in $(aclocaldir).
-OLD_M4_FILES = \
-  build-to-host.m4 \
-  codeset.m4 \
-  extern-inline.m4 \
-  fcntl-o.m4 \
-  gettext.m4 \
-  glibc2.m4 \
-  glibc21.m4 \
-  host-cpu-c-abi.m4 \
-  iconv.m4 \
-  intdiv0.m4 \
-  intl.m4 \
-  intldir.m4 \
-  intlmacosx.m4 \
-  intmax.m4 \
-  inttypes.m4 \
-  inttypes-h.m4 \
-  inttypes-pri.m4 \
-  inttypes_h.m4 \
-  isc-posix.m4 \
-  lcmessage.m4 \
-  lib-ld.m4 \
-  lib-link.m4 \
-  lib-prefix.m4 \
-  lock.m4 \
-  longdouble.m4 \
-  longlong.m4 \
-  nls.m4 \
-  po.m4 \
-  printf-posix.m4 \
-  progtest.m4 \
-  signed.m4 \
-  size_max.m4 \
-  stdint_h.m4 \
-  threadlib.m4 \
-  uintmax_t.m4 \
-  ulonglong.m4 \
-  visibility.m4 \
-  wchar_t.m4 \
-  wint_t.m4 \
-  xsize.m4
-
 all: all-am
 
 .SUFFIXES:
@@ -3876,48 +3749,48 @@
 
 clean-libtool:
 	-rm -rf .libs _libs
-install-examplesbuildauxDATA: $(examplesbuildaux_DATA)
+install-aclocalDATA: $(aclocal_DATA)
 	@$(NORMAL_INSTALL)
-	@list='$(examplesbuildaux_DATA)'; test -n "$(examplesbuildauxdir)" || list=; \
+	@list='$(aclocal_DATA)'; test -n "$(aclocaldir)" || list=; \
 	if test -n "$$list"; then \
-	  echo " $(MKDIR_P) '$(DESTDIR)$(examplesbuildauxdir)'"; \
-	  $(MKDIR_P) "$(DESTDIR)$(examplesbuildauxdir)" || exit 1; \
+	  echo " $(MKDIR_P) '$(DESTDIR)$(aclocaldir)'"; \
+	  $(MKDIR_P) "$(DESTDIR)$(aclocaldir)" || exit 1; \
 	fi; \
 	for p in $$list; do \
 	  if test -f "$$p"; then d=; else d="$(srcdir)/"; fi; \
 	  echo "$$d$$p"; \
 	done | $(am__base_list) | \
 	while read files; do \
-	  echo " $(INSTALL_DATA) $$files '$(DESTDIR)$(examplesbuildauxdir)'"; \
-	  $(INSTALL_DATA) $$files "$(DESTDIR)$(examplesbuildauxdir)" || exit $$?; \
+	  echo " $(INSTALL_DATA) $$files '$(DESTDIR)$(aclocaldir)'"; \
+	  $(INSTALL_DATA) $$files "$(DESTDIR)$(aclocaldir)" || exit $$?; \
 	done
 
-uninstall-examplesbuildauxDATA:
+uninstall-aclocalDATA:
 	@$(NORMAL_UNINSTALL)
-	@list='$(examplesbuildaux_DATA)'; test -n "$(examplesbuildauxdir)" || list=; \
+	@list='$(aclocal_DATA)'; test -n "$(aclocaldir)" || list=; \
 	files=`for p in $$list; do echo $$p; done | sed -e 's|^.*/||'`; \
-	dir='$(DESTDIR)$(examplesbuildauxdir)'; $(am__uninstall_files_from_dir)
-install-macrosDATA: $(macros_DATA)
+	dir='$(DESTDIR)$(aclocaldir)'; $(am__uninstall_files_from_dir)
+install-examplesbuildauxDATA: $(examplesbuildaux_DATA)
 	@$(NORMAL_INSTALL)
-	@list='$(macros_DATA)'; test -n "$(macrosdir)" || list=; \
+	@list='$(examplesbuildaux_DATA)'; test -n "$(examplesbuildauxdir)" || list=; \
 	if test -n "$$list"; then \
-	  echo " $(MKDIR_P) '$(DESTDIR)$(macrosdir)'"; \
-	  $(MKDIR_P) "$(DESTDIR)$(macrosdir)" || exit 1; \
+	  echo " $(MKDIR_P) '$(DESTDIR)$(examplesbuildauxdir)'"; \
+	  $(MKDIR_P) "$(DESTDIR)$(examplesbuildauxdir)" || exit 1; \
 	fi; \
 	for p in $$list; do \
 	  if test -f "$$p"; then d=; else d="$(srcdir)/"; fi; \
 	  echo "$$d$$p"; \
 	done | $(am__base_list) | \
 	while read files; do \
-	  echo " $(INSTALL_DATA) $$files '$(DESTDIR)$(macrosdir)'"; \
-	  $(INSTALL_DATA) $$files "$(DESTDIR)$(macrosdir)" || exit $$?; \
+	  echo " $(INSTALL_DATA) $$files '$(DESTDIR)$(examplesbuildauxdir)'"; \
+	  $(INSTALL_DATA) $$files "$(DESTDIR)$(examplesbuildauxdir)" || exit $$?; \
 	done
 
-uninstall-macrosDATA:
+uninstall-examplesbuildauxDATA:
 	@$(NORMAL_UNINSTALL)
-	@list='$(macros_DATA)'; test -n "$(macrosdir)" || list=; \
+	@list='$(examplesbuildaux_DATA)'; test -n "$(examplesbuildauxdir)" || list=; \
 	files=`for p in $$list; do echo $$p; done | sed -e 's|^.*/||'`; \
-	dir='$(DESTDIR)$(macrosdir)'; $(am__uninstall_files_from_dir)
+	dir='$(DESTDIR)$(examplesbuildauxdir)'; $(am__uninstall_files_from_dir)
 tags TAGS:
 
 ctags CTAGS:
@@ -3961,7 +3834,7 @@
 check: check-am
 all-am: Makefile $(DATA)
 installdirs:
-	for dir in "$(DESTDIR)$(examplesbuildauxdir)" "$(DESTDIR)$(macrosdir)"; do \
+	for dir in "$(DESTDIR)$(aclocaldir)" "$(DESTDIR)$(examplesbuildauxdir)"; do \
 	  test -z "$$dir" || $(MKDIR_P) "$$dir"; \
 	done
 install: install-am
@@ -4014,8 +3887,7 @@
 
 info-am:
 
-install-data-am: install-data-local install-examplesbuildauxDATA \
-	install-macrosDATA
+install-data-am: install-aclocalDATA install-examplesbuildauxDATA
 
 install-dvi: install-dvi-am
 
@@ -4059,37 +3931,27 @@
 
 ps-am:
 
-uninstall-am: uninstall-examplesbuildauxDATA uninstall-local \
-	uninstall-macrosDATA
+uninstall-am: uninstall-aclocalDATA uninstall-examplesbuildauxDATA
 
 .MAKE: install-am install-strip
 
 .PHONY: all all-am check check-am clean clean-generic clean-libtool \
 	cscopelist-am ctags-am distclean distclean-generic \
 	distclean-libtool distdir dvi dvi-am html html-am info info-am \
-	install install-am install-data install-data-am \
-	install-data-local install-dvi install-dvi-am \
+	install install-aclocalDATA install-am install-data \
+	install-data-am install-dvi install-dvi-am \
 	install-examplesbuildauxDATA install-exec install-exec-am \
 	install-html install-html-am install-info install-info-am \
-	install-macrosDATA install-man install-pdf install-pdf-am \
-	install-ps install-ps-am install-strip installcheck \
-	installcheck-am installdirs maintainer-clean \
-	maintainer-clean-generic mostlyclean mostlyclean-generic \
-	mostlyclean-libtool pdf pdf-am ps ps-am tags-am uninstall \
-	uninstall-am uninstall-examplesbuildauxDATA uninstall-local \
-	uninstall-macrosDATA
+	install-man install-pdf install-pdf-am install-ps \
+	install-ps-am install-strip installcheck installcheck-am \
+	installdirs maintainer-clean maintainer-clean-generic \
+	mostlyclean mostlyclean-generic mostlyclean-libtool pdf pdf-am \
+	ps ps-am tags-am uninstall uninstall-aclocalDATA uninstall-am \
+	uninstall-examplesbuildauxDATA
 
 .PRECIOUS: Makefile
 
 
-# We don't install the *.m4 files in $(aclocaldir) any more.
-install-data-local: uninstall-macros
-uninstall-local: uninstall-macros
-uninstall-macros:
-	for f in $(OLD_M4_FILES); do \
-	  rm -f '$(DESTDIR)@aclocaldir@'/$$f; \
-	done
-
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
diff -Naur a/gettext-tools/misc/gettextize.in b/gettext-tools/misc/gettextize.in
--- a/gettext-tools/misc/gettextize.in	2025-04-02 14:07:39.000000000 +0200
+++ b/gettext-tools/misc/gettextize.in	2025-02-25 10:49:44.000000000 +0100
@@ -886,7 +886,7 @@
   for file in $m4filelist; do
     same=no
     if test -f "$srcdir/$m4dir/$file"; then
-      if cmp -s "@datadir@/gettext/m4/$file" "$srcdir/$m4dir/$file"; then
+      if cmp -s "@datadir@/aclocal/$file" "$srcdir/$m4dir/$file"; then
         same=yes
       fi
     else
@@ -900,7 +900,7 @@
       fi
     fi
     func_backup "$m4dir/$file"
-    func_linkorcopy "@datadir@/gettext/m4/$file" "@datadir@/gettext/m4/$file" "$m4dir/$file"
+    func_linkorcopy "@datadir@/aclocal/$file" "@datadir@/aclocal/$file" "$m4dir/$file"
   done
   missing_m4Makefileam=
   if test -n "$added_m4files"; then
@@ -1092,7 +1092,7 @@
   please="$please
 Please add the files
 $m4filelist
-from the @datadir@/gettext/m4 directory to your aclocal.m4 file.
+from the @datadir@/aclocal directory to your aclocal.m4 file.
 "
 fi
 
