diff -Naur a/gettext-tools/src/msgl-cat.c b/gettext-tools/src/msgl-cat.c
--- a/gettext-tools/src/msgl-cat.c	2023-09-19 04:11:23.000000000 +0200
+++ b/gettext-tools/src/msgl-cat.c	2024-08-19 18:39:09.257349406 +0200
@@ -309,6 +309,7 @@
                                    mp->msgid);
                       multiline_error (xstrdup (""),
                                        xasprintf ("%s\n", errormsg));
+                      free (errormsg);
                     }
                 }
               else
diff -Naur a/gettext-tools/src/x-csharp.c b/gettext-tools/src/x-csharp.c
--- a/gettext-tools/src/x-csharp.c	2023-09-18 22:09:26.000000000 +0200
+++ b/gettext-tools/src/x-csharp.c	2024-08-19 18:40:18.913629426 +0200
@@ -564,6 +564,7 @@
   buffer[buflen] = '\0';
   savable_comment_add (buffer);
   lexical_context = lc_outside;
+  free (buffer);
 }
 
 
diff -Naur a/gettext-tools/src/xg-arglist-callshape.c b/gettext-tools/src/xg-arglist-callshape.c
--- a/gettext-tools/src/xg-arglist-callshape.c	2019-05-11 13:29:32.000000000 +0200
+++ b/gettext-tools/src/xg-arglist-callshape.c	2024-08-19 18:40:47.053742604 +0200
@@ -118,6 +118,7 @@
                   memcpy (xcomment, xcomment_start, xcomment_len);
                   xcomment[xcomment_len] = '\0';
                   string_list_append (&xcomments, xcomment);
+                  free (xcomment);
                 }
               else
                 break;
diff -Naur a/gettext-tools/src/x-java.c b/gettext-tools/src/x-java.c
--- a/gettext-tools/src/x-java.c	2023-09-18 22:09:26.000000000 +0200
+++ b/gettext-tools/src/x-java.c	2024-08-19 18:40:47.049742587 +0200
@@ -461,6 +461,7 @@
     --buflen;
   buffer[buflen] = '\0';
   savable_comment_add (buffer);
+  free (buffer);
 }
 
 
diff -Naur a/gettext-tools/src/x-rst.c b/gettext-tools/src/x-rst.c
--- a/gettext-tools/src/x-rst.c	2022-09-11 03:12:41.000000000 +0200
+++ b/gettext-tools/src/x-rst.c	2024-08-19 18:40:47.053742604 +0200
@@ -651,6 +651,8 @@
           else
             goto invalid_rsj;
 
+          free (s1);
+
           /* Parse a comma.  */
           c = phase2_getc ();
           if (c == '}')
@@ -667,6 +669,7 @@
 
   fp = NULL;
   real_file_name = NULL;
+  free (logical_file_name);
   logical_file_name = NULL;
   line_number = 0;
 
