diff -Naur a/configure.ac b/configure.ac
--- a/configure.ac	2023-12-24 18:42:02.000000000 +0100
+++ b/configure.ac	2023-12-25 22:35:05.481018055 +0100
@@ -2713,12 +2713,7 @@
   CFLAGS="$old_CFLAGS"
 fi
 
-AC_PATH_PROGS(FCCACHE, fc-cache, false)
-if test x"$FCCACHE" = "xfalse"; then
-  AM_CONDITIONAL(HAVE_FC_CACHE, false)
-else
-  AM_CONDITIONAL(HAVE_FC_CACHE, true)
-fi
+AM_CONDITIONAL(HAVE_FC_CACHE, false)
 
 dsound_header_no_lib="no"
 
