diff -Naur a/protocol/hp_ipp.h b/protocol/hp_ipp.h
--- a/protocol/hp_ipp.h	2025-02-27 13:12:29.000000000 +0100
+++ b/protocol/hp_ipp.h	2025-05-23 01:12:44.118538949 +0200
@@ -151,6 +151,7 @@
 http_t*    acquireCupsInstance();
 const char *getCupsErrorString(int status);
 void freePrinterList(printer_t *list);
+void _releaseCupsInstance();
  
 static ssize_t raw_ipp_response_read_callback(raw_ipp  *raw_buffer,ipp_uchar_t *buffer,size_t length);
 static ssize_t raw_ipp_request_callback(volatile raw_ipp  *raw_buffer, ipp_uchar_t *buffer, size_t length);
@@ -163,6 +164,10 @@
 ipp_t * networkDoRequest(ipp_t *request, char* device_uri,const char *resource);
 ipp_t * getDeviceStatusAttributes(char* device_uri,char* printer_name, int *count);
 int     getCupsPrinters(printer_t **printer_list);
+int     addCupsPrinter(char *name, char *device_uri, char *location, char *ppd_file, char *model, char *info);
+int     delCupsPrinter(char *pr_name);
+int     setDefaultCupsPrinter(char *pr_name);
+int     controlCupsPrinter(char *pr_name, int op);
 
 HPIPP_RESULT parseResponseHeader(char* header, int *content_length, int *chunked, int* header_size);
 HPIPP_RESULT prepend_http_header(raw_ipp *raw_request,const char *resource);
