diff -Naur a/scan/sane/bb_ledm.c b/scan/sane/bb_ledm.c
--- a/scan/sane/bb_ledm.c	2025-02-27 13:12:29.000000000 +0100
+++ b/scan/sane/bb_ledm.c	2025-05-23 01:27:36.042257531 +0200
@@ -28,6 +28,7 @@
 # include <stdlib.h>
 
 # include <stdint.h>
+# include <unistd.h>
 
 # define _STRINGIZE(x) #x
 # define STRINGIZE(x) _STRINGIZE(x)
diff -Naur a/scan/sane/http.c b/scan/sane/http.c
--- a/scan/sane/http.c	2025-02-27 13:12:29.000000000 +0100
+++ b/scan/sane/http.c	2025-05-23 01:26:28.989983134 +0200
@@ -36,6 +36,7 @@
 #include <string.h>
 #include <syslog.h>
 #include <ctype.h>
+#include <unistd.h>
 #include "hpmud.h"
 #include "http.h"
 
diff -Naur a/scan/sane/sclpml.c b/scan/sane/sclpml.c
--- a/scan/sane/sclpml.c	2025-02-27 13:12:29.000000000 +0100
+++ b/scan/sane/sclpml.c	2025-05-23 01:28:53.238572977 +0200
@@ -30,6 +30,7 @@
 #include <unistd.h>
 #include <fcntl.h>
 #include <string.h>
+#include <sys/time.h>
 #include "sane.h"
 #include "saneopts.h"
 #include "common.h"
