diff -ru poedit-3.6.1.orig/src/subprocess.cpp poedit-3.6.1/src/subprocess.cpp
--- poedit-3.6.1.orig/src/subprocess.cpp	2025-06-07 11:33:03.472979413 +0200
+++ poedit-3.6.1/src/subprocess.cpp	2025-06-07 11:47:59.128297240 +0200
@@ -171,7 +171,7 @@
 
 wxString Arguments::pretty_print() const
 {
-    return "[" + boost::join(m_args, ", ") + "]";
+    return str::to_wstring("[") + boost::join(m_args, ", ") + str::to_wstring("]");
 }
 
 
