diff -Naur a/lib/dpkg/dpkg.h b/lib/dpkg/dpkg.h
--- a/lib/dpkg/dpkg.h	2025-03-09 19:01:08.000000000 +0100
+++ b/lib/dpkg/dpkg.h	2025-03-18 02:02:56.940144721 +0100
@@ -52,7 +52,7 @@
 #define MAXCONFFILENAME     1000
 #define MAXDIVERTFILENAME   1024
 #define MAXCONTROLFILENAME  100
-#define DEBEXT             ".deb"
+#define DEBEXT             ".uhu"
 #define REMOVECONFFEXTS    "~", ".bak", "%", \
                            DPKGTEMPEXT, DPKGNEWEXT, DPKGOLDEXT, DPKGDISTEXT
 
diff -Naur a/po/hu.po b/po/hu.po
--- a/po/hu.po	2025-03-09 19:01:08.000000000 +0100
+++ b/po/hu.po	2025-03-18 02:02:56.948144756 +0100
@@ -4598,8 +4598,8 @@
 msgstr "--%s --recursive legalább 1 ösvény argumentumot vár"
 
 #: src/main/archives.c
-msgid "searched, but found no packages (files matching *.deb)"
-msgstr "kerestem, de nem leltem csomagokat (*.deb fájlokat)"
+msgid "searched, but found no packages (files matching *.uhu)"
+msgstr "kerestem, de nem leltem csomagokat (*.uhu fájlokat)"
 
 #: src/main/archives.c
 #, c-format
diff -Naur a/src/main/archives.c b/src/main/archives.c
--- a/src/main/archives.c	2025-03-09 19:01:08.000000000 +0100
+++ b/src/main/archives.c	2025-03-18 02:02:56.948144756 +0100
@@ -1546,7 +1546,7 @@
     }
 
     if (!nfiles)
-      ohshit(_("searched, but found no packages (files matching *.deb)"));
+      ohshit(_("searched, but found no packages (files matching *.uhu)"));
 
     arglist[nfiles] = NULL;
     argp = (const char **volatile)arglist;
