From eb4508b2e6135e0762e93dc8b6fd6258748e93d2 Mon Sep 17 00:00:00 2001
From: rezso <rezso@rezso.net>
Date: Sun, 20 Jul 2025 00:48:37 +0200
Subject: change to rezso.net


diff --git a/src/build/build b/src/build/build
index 28a7da8..0f7ca73 100755
--- a/src/build/build
+++ b/src/build/build
@@ -19,15 +19,13 @@ scrollout
 # Páder Rezső <rezso@rezso.net>
 
 BUILD_VERSION="1.4.0"
-VENDOR="ubk.hu"
-PACKAGER=${PACKAGER:=UBK <ubk@ubk.hu>}
-BUILD_GPGKEY=${BUILD_GPGKEY:=UBK}
+VENDOR="rezso.net"
+PACKAGER=${PACKAGER:=Páder Rezső <rezso@rezso.net>}
+BUILD_GPGKEY=${BUILD_GPGKEY:=D1F82444B1ACA1CD}
 
 case `basename $0` in
 	build)
-		export UHUBUILD_DISTR=UBK4;;
-	build-ubk3)
-		export UHUBUILD_DISTR=UBK3;;
+		export UHUBUILD_DISTR=UBK5;;
 	build-ubk4)
 		export UHUBUILD_DISTR=UBK4;;
 	build-ubk5)
