From 05d6a0528248051ac53027fdb384b1dc0047ba8b Mon Sep 17 00:00:00 2001
From: rezso <rezso@rezso.net>
Date: Mon, 18 Aug 2025 23:58:56 +0200
Subject: cleanup


diff --git a/src/build/build b/src/build/build
index 0f7ca73..a752db9 100755
--- a/src/build/build
+++ b/src/build/build
@@ -34,10 +34,7 @@ esac
 
 UHUBUILD_DISTR_NUM=${UHUBUILD_DISTR/UBK/}
 
-[ "$UID" != 0 ] && {
-	redecho "A buildszkript csak root-ként futtatható!"
-	exit 1
-	}
+dieifuser
 
 export UB_ARCH="amd64"
 export UHUBUILD_VAR=/var/uhubuild
diff --git a/src/build/build-all b/src/build/build-all
index 560136a..d86e352 100755
--- a/src/build/build-all
+++ b/src/build/build-all
@@ -6,11 +6,8 @@
 # Összeállította: Sári Gábor <saga@kde.hu>
 # 2004. január 19. táján a II. UHU Csomagoló Party után
 
-[ "$UID" != 0 ] && {
-	echo "Csak a rendszergazdaként indítható ez a szkript!"
-	exit 1
-}
+dieifuser
 
-for i in ubk3 ubk4 ubk5; do
+for i in ubk4 ubk5; do
 	/usr/bin/build-"$i" "$@"
 done
diff --git a/src/build/build-clean b/src/build/build-clean
index 6aeee12..e28749f 100755
--- a/src/build/build-clean
+++ b/src/build/build-clean
@@ -17,19 +17,14 @@ arch="amd64"
 
 case ${BASENAME} in
 	build-clean)
-		export DISTR=UBK4;;
-	build-clean-ubk3)
-		export DISTR=UBK3;;
+		export DISTR=UBK5;;
 	build-clean-ubk4)
 		export DISTR=UBK4;;
 	build-clean-ubk5)
 		export DISTR=UBK5;;
 esac
 
-[ "$UID" != 0 ] && {
-	redecho "A ${BASENAME} csak rendszergazdai jogosultságokkal futtatható!"
-	exit 1
-}
+dieifuser
 
 function help() {
 	appleecho "${BASENAME} ${BUILDCLEAN_VERSION}"
diff --git a/src/build/build-clean-all b/src/build/build-clean-all
index a4831f8..9f84bd9 100755
--- a/src/build/build-clean-all
+++ b/src/build/build-clean-all
@@ -4,11 +4,8 @@
 #
 # A szkriptet eredetileg készítette: Charlos, 2007. március 9.-én.
 
-[ "$UID" != 0 ] && {
-	echo "Csak a rendszergazdaként indítható ez a szkript!"
-	exit 1
-}
+dieifuser
 
-for i in ubk3 ubk4 ubk5; do
+for i in ubk4 ubk5; do
 	/usr/bin/build-clean-"$i" "$@"
 done
diff --git a/src/common b/src/common
index dbcc1e3..312e7fa 100644
--- a/src/common
+++ b/src/common
@@ -148,28 +148,28 @@ diepoint ()
 # -----------------------------------------------------------------------------
 warnifroot ()
 {
-	if [ "$UID" == 0 ]; then
+	if [ $(id -u) -eq 0 ]; then
 		warn "Ne root-ként indíts!"
 	fi
 }
 
 warnifuser ()
 {
-	if [ "$UID" != 0 ]; then
+	if [ $(id -u) -ne 0 ]; then
 		warn "Root-ként indíts!"
 	fi
 }
 
 dieifroot ()
 {
-	if [ "$UID" == 0 ]; then
+	if [ $(id -u) -eq 0 ]; then
 		die "Ne root-ként indíts!"
 	fi
 }
 
 dieifuser ()
 {
-	if [ "$UID" != 0 ]; then
+	if [ $(id -u) -ne 0 ]; then
 		die "Root-ként indíts!"
 	fi
 }
diff --git a/src/uhubuild-pack b/src/uhubuild-pack
index a009a1d..0981407 100755
--- a/src/uhubuild-pack
+++ b/src/uhubuild-pack
@@ -18,7 +18,7 @@ chmod -R u+w DEBIAN
 
 if [ ! -f data.tar.gz ]; then
 
-if [ "$UID" != 0 ]; then
+if [ $(id -u) -ne 0 ]; then
 	echo 'Figyelem! Nem root vagy!' >&2
 	echo 'A készülő csomagban a fájlok tulajdonosa minden bizonnyal nem root lesz.' >&2
 	echo 'Válts át root felhasználóvá, vagy sima felhasználóként használd' >&2
