From 1b8676c32392875d7fd6fffafa8f437c9ded5722 Mon Sep 17 00:00:00 2001
From: rezso <rezso@rezso.net>
Date: Sun, 17 Aug 2025 23:42:13 +0200
Subject: drop unused scripts


diff --git a/doc/uhubuild.sgml b/doc/uhubuild.sgml
index d5be614..8c5a03c 100644
--- a/doc/uhubuild.sgml
+++ b/doc/uhubuild.sgml
@@ -442,24 +442,9 @@ semmiképp se saját kézzel-lábbal készített .tar.gz, mert az könnyen lehet
 hogy nem jó.
 
 
-<sect1>uhubuild-extract<p>
-
-Kábé az uhubuild-pack ellentéte. A csomagot kipakolja az aktuális könyvtár
-alá. Ajánlott root-ként, vagy a pretendroot könyvtáron keresztül futtatni,
-hogy a tulajok megmaradjanak.
-
-
-<sect1>uhubuild-expand<p>
-
-Mint az uhubuild-extract, de a data.tar.gz-t nem tömöríti ki. Így jóval
-gyorsabb lesz újra becsomagolni, viszont csak a DEBIAN könyvtár fájljain
-tudunk változtatni, a csomag igazi fájljain nem.
-
-
 
 <sect>Környezeti változók<p>
 
-
 Az uhubuild rendszer működését kívülről olyan környezeti változókkal
 szabályozzuk, melyek neve UHUBUILD_ kezdetű.
 
diff --git a/src/uhubuild-expand b/src/uhubuild-expand
deleted file mode 100755
index b1c29e3..0000000
--- a/src/uhubuild-expand
+++ /dev/null
@@ -1,10 +0,0 @@
-#!/bin/bash -e
-
-. /usr/lib/uhubuild/uhubuild-common
-
-if [ $# != 1 ]; then
-	die 'uhubuild-expand: Szükségem van argumentumra!'
-fi
-
-dpkg-deb --control "$1"
-ar x "$1" data.tar.gz
diff --git a/src/uhubuild-extract b/src/uhubuild-extract
deleted file mode 100755
index 32142d9..0000000
--- a/src/uhubuild-extract
+++ /dev/null
@@ -1,18 +0,0 @@
-#!/bin/bash -e
-
-. /usr/lib/uhubuild/uhubuild-common
-
-if [ $# != 1 ]; then
-	die 'uhubuild-extract: Szükségem van argumentumra!'
-fi
-
-if [ "$UID" != "0" ]; then
-	echo 'Figyelem! Nem root vagy!' >&2
-	echo 'A kicsomagolás során elvész a fájlok tulajdonosa és csoportja.' >&2
-	echo 'Ha újra össze szeretnéd csomagolni a fájlokat egy uhu csomaggá,' >&2
-	echo 'válts át root felhasználóvá, vagy sima felhasználóként használd' >&2
-	echo 'a pretendroot program által nyújtott virtuális root környezetet.' >&2
-fi
-
-dpkg-deb --control "$1"
-dpkg-deb --extract "$1" .
diff --git a/src/uhubuild-pack b/src/uhubuild-pack
index 954decd..a009a1d 100755
--- a/src/uhubuild-pack
+++ b/src/uhubuild-pack
@@ -18,7 +18,7 @@ chmod -R u+w DEBIAN
 
 if [ ! -f data.tar.gz ]; then
 
-if [ "$UID" != "0" ]; then
+if [ "$UID" != 0 ]; then
 	echo 'Figyelem! Nem root vagy!' >&2
 	echo 'A készülő csomagban a fájlok tulajdonosa minden bizonnyal nem root lesz.' >&2
 	echo 'Válts át root felhasználóvá, vagy sima felhasználóként használd' >&2
