From d0cc77b8b3a0817bc397683434cd5ac085e26c81 Mon Sep 17 00:00:00 2001
From: rezso <rezso@rezso.net>
Date: Tue, 21 Nov 2023 00:59:16 +0100
Subject: pack: follow dpkg change


diff --git a/src/steps/pack b/src/steps/pack
index b342ea8..27054b4 100755
--- a/src/steps/pack
+++ b/src/steps/pack
@@ -32,6 +32,6 @@ for pkg in $UB_PACKAGES; do
 	pretendroot \
 	  uhubuild-pack "$UB_PACKAGEDIR/$pkg" "$UB_RESULTDIR"
 done
-for pkg in "$UB_RESULTDIR"/*.ubk.uhu; do
-	mv "$pkg" "${pkg%.ubk.uhu}.${DIST}.uhu"
+for pkg in "$UB_RESULTDIR"/*.uhu; do
+	mv "$pkg" "${pkg%.uhu}.${DIST}.uhu"
 done
