From 39ea0b424c01f5bbc38e870b287fc59a111b90ce Mon Sep 17 00:00:00 2001
From: rezso <rezso@rezso.net>
Date: Fri, 5 Jan 2024 05:49:54 +0100
Subject: szimlink "könyvtárak" kezelése
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit


diff --git a/src/steps/fhs b/src/steps/fhs
index e78629f..739a77d 100755
--- a/src/steps/fhs
+++ b/src/steps/fhs
@@ -86,6 +86,16 @@ for pkg in $UB_PACKAGES; do
 			error "$dir könyvtár? Ejnye! Ilyet nem szabad!"
 		fi
 	done
+	# Ezek csak szimlinkek, nem valós könyvtárak
+	for dir in \
+	  /bin \
+	  /lib \
+	  /lib64 \
+	  /sbin; do
+		if [ -d "$pkg/$dir" ]; then
+			error "$dir csak egy szimlink, nem valós könyvtár!"
+		fi
+	done
 	while read file; do
 		error "${file#$pkg/}? Ejnye! Ilyet nem szabad!"
 	done < <(find "$pkg" -mindepth 1 -maxdepth 1 -type f)
diff --git a/src/steps/install b/src/steps/install
index 51acce6..a75fb5b 100755
--- a/src/steps/install
+++ b/src/steps/install
@@ -18,7 +18,7 @@ if [ -f "$UB_SRCDIR/install" ]; then
 	# Az alábbi listát tartsd szinkronban a split szkriptben lévővel!
 	pretendroot \
 	  mkdir -p \
-	    bin etc/{pam.d,sysconfig} lib sbin \
+	    etc/{pam.d,sysconfig} \
 	    usr/{bin,include,lib,libexec,sbin} \
 	    usr/share/{applications,doc,info,locale,pixmaps} \
 	    usr/share/man/man{1,2,3,4,5,6,7,8,9,n} \
diff --git a/src/steps/split b/src/steps/split
index fab011b..4152d60 100755
--- a/src/steps/split
+++ b/src/steps/split
@@ -145,7 +145,7 @@ for pkg in $UB_PACKAGES; do
 	# Az alábbi listát tartsd szinkronban az install szkriptben lévővel!
 	pretendroot \
 	  rmdir -p --ignore-fail-on-non-empty \
-	    bin etc/{pam.d,sysconfig} lib sbin \
+	    etc/{pam.d,sysconfig} \
 	    usr/{bin,include,lib,libexec,sbin} \
 	    usr/share/{applications,doc,info,locale,pixmaps} \
 	    usr/share/man/man{1,2,3,4,5,6,7,8,9,n} \
diff --git a/src/uhubuild-u2d.sh b/src/uhubuild-u2d.sh
index 7827f31..7bf0e66 100644
--- a/src/uhubuild-u2d.sh
+++ b/src/uhubuild-u2d.sh
@@ -175,7 +175,6 @@ function u2dxfce {
     urllist https://archive.xfce.org/src/$subdir/$project/"$ver/" | splitver '^'$project'-([0-9.]+).tar.(gz|bz2|xz)$' | sort -V | tail -n 1
 }
 
-
 # script for perl modules
 function u2dcpan() {
 	ver="$(curl -m 10 -s http://cpanmetadb.plackperl.org/v1.0/package/$1)"
