From ef284b42237786c8a4704e4ce7e0fcd41297027c Mon Sep 17 00:00:00 2001
From: rezso <rezso@rezso.net>
Date: Wed, 24 Jul 2024 21:34:12 +0200
Subject: javítások
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit


diff --git a/src/dl/get b/src/dl/get
index 7788944..7329a96 100755
--- a/src/dl/get
+++ b/src/dl/get
@@ -68,12 +68,12 @@ function _get() {
 		fi
 
 		if [ -z "$http_proxy" ]; then
-			curl -qfLNk --max-redir 8 --user "$user" --disable-epsv "${limit[@]}" "${connecttimeout[@]}" "${maxtime[@]}" "${speedlimit[@]}" "${speedtime[@]}" "$url" >"$tmpfile" || return
+			curl -qfLNk --max-redirs 8 --user "$user" --disable-epsv "${limit[@]}" "${connecttimeout[@]}" "${maxtime[@]}" "${speedlimit[@]}" "${speedtime[@]}" "$url" >"$tmpfile" || return
 		else
 			if [ -z "$http_proxy_user" ]; then
-				curl -qfLNk --proxy "$http_proxy" --max-redir 8 --user "$user" --disable-epsv "${limit[@]}" "${connecttimeout[@]}" "${maxtime[@]}" "${speedlimit[@]}" "${speedtime[@]}" "$url" >"$tmpfile" || return
+				curl -qfLNk --proxy "$http_proxy" --max-redirs 8 --user "$user" --disable-epsv "${limit[@]}" "${connecttimeout[@]}" "${maxtime[@]}" "${speedlimit[@]}" "${speedtime[@]}" "$url" >"$tmpfile" || return
 			else
-				curl -qfLNk --proxy "$http_proxy" --proxy-user "$http_proxy_user" --max-redir 8 --user "$user" --disable-epsv "${limit[@]}" "${connecttimeout[@]}" "${maxtime[@]}" "${speedlimit[@]}" "${speedtime[@]}" "$url" >"$tmpfile" || return
+				curl -qfLNk --proxy "$http_proxy" --proxy-user "$http_proxy_user" --max-redirs 8 --user "$user" --disable-epsv "${limit[@]}" "${connecttimeout[@]}" "${maxtime[@]}" "${speedlimit[@]}" "${speedtime[@]}" "$url" >"$tmpfile" || return
 			fi
 		fi
 
@@ -108,12 +108,12 @@ function _get() {
 		fi
 
 		if [ -z "$http_proxy" ]; then
-			curl -qfLNk --max-redir 8 "${limit[@]}" "${connecttimeout[@]}" "${maxtime[@]}" "${speedlimit[@]}" "${speedtime[@]}" "$1" >"$tmpfile" || return
+			curl -qfLNk --max-redirs 8 "${limit[@]}" "${connecttimeout[@]}" "${maxtime[@]}" "${speedlimit[@]}" "${speedtime[@]}" "$1" >"$tmpfile" || return
 		else
 			if [ -z "$http_proxy_user" ]; then
-				curl -qfLNk --proxy "$http_proxy" --max-redir 8 "${limit[@]}" "${connecttimeout[@]}" "${maxtime[@]}" "${speedlimit[@]}" "${speedtime[@]}" "$1" >"$tmpfile" || return
+				curl -qfLNk --proxy "$http_proxy" --max-redirs 8 "${limit[@]}" "${connecttimeout[@]}" "${maxtime[@]}" "${speedlimit[@]}" "${speedtime[@]}" "$1" >"$tmpfile" || return
 			else
-				curl -qfLNk --proxy "$http_proxy" --proxy-user "$http_proxy_user" --max-redir 8 "${limit[@]}" "${connecttimeout[@]}" "${maxtime[@]}" "${speedlimit[@]}" "${speedtime[@]}" "$1" >"$tmpfile" || return
+				curl -qfLNk --proxy "$http_proxy" --proxy-user "$http_proxy_user" --max-redirs 8 "${limit[@]}" "${connecttimeout[@]}" "${maxtime[@]}" "${speedlimit[@]}" "${speedtime[@]}" "$1" >"$tmpfile" || return
 			fi
 		fi
 
diff --git a/src/steps/depends b/src/steps/depends
index 5a03490..b92f43b 100755
--- a/src/steps/depends
+++ b/src/steps/depends
@@ -485,6 +485,18 @@ for pkg in $UB_PACKAGES; do
 		fi
 	done
 
+# python: nem mindig van #!/usr/bin/python* a .py fájlok elején...
+    for pv in 3.12 3.11 3.10 3.9 3.8 3.7; do
+        if [ -d ./usr/lib/python$pv ]; then
+            echo "python3" >> "$UB_TMPDIR/pkg-noversion"
+        fi
+    done
+
+# perl
+    if [ -d ./usr/lib/perl5 ]; then
+        echo "perl" >> "$UB_TMPDIR/pkg-noversion"
+    fi
+
 # Most a file-* fájlokban fájlnevek vannak. Ebből kell csinálni csomagneveket.
 
 	sort -u < "$UB_TMPDIR/file-atleast" > "$UB_TMPDIR/file-atleast.tmp"
@@ -503,18 +515,6 @@ for pkg in $UB_PACKAGES; do
 		echo " $pkg: automatikus dependencia keresés fázis kihagyva"
 	fi # skip-autodepend
 
-# python: nem mindig van #!/usr/bin/python* a .py fájlok elején...
-    for pv in 3.12 3.11 3.10 3.9 3.8 3.7; do
-        if [ -d ./usr/lib/python$pv ]; then
-            echo "python3" >> "$UB_TMPDIR/pkg-noversion"
-        fi
-    done
-
-# perl
-    if [ -d ./usr/lib/perl5 ]; then
-        echo "perl" >> "$UB_TMPDIR/pkg-noversion"
-    fi
-
 # Az uhu-pkg majnem mindenkinek kell a postinst szkript miatt.
 	if ! grep -qx skip-postinst "$UB_SRCDIR/packages/$pkg/options" || \
 	   ! grep -qx skip-prerm    "$UB_SRCDIR/packages/$pkg/options" || \
