From e24fed8d2c34dfb427ca8777a945fb195297d41d Mon Sep 17 00:00:00 2001
From: rezso <rezso@rezso.net>
Date: Mon, 19 Aug 2024 18:08:43 +0200
Subject: macros update


diff --git a/config/macros b/config/macros
index c5a5906..96cec76 100644
--- a/config/macros
+++ b/config/macros
@@ -1,2 +1,2 @@
-%_ autoconf autoconf-archive automake bash bc binutils bison bzip2 bzip2-dev ccache cmake coreutils cracklib desktop-file-utils diffutils dpkg file findutils flex flex-dev gawk gcc gettext gettext-tools glibc glibc-dev grep gzip inetutils kernel-headers libtool licenses m4 make meson ninja optipng pam pam-dev patch pkgconfig pretendroot sed shadow tar uhu-base uhubuild-here uhu-pkg which xz-dev zlib-dev zstd-dev
+%_ autoconf autoconf-archive automake bash bc binutils bison bzip2 bzip2-dev ccache cmake coreutils cracklib desktop-file-utils diffutils dpkg file findutils flex flex-dev gawk gcc gettext-dev gettext-tools glibc glibc-dev grep gzip inetutils kernel-headers libtool licenses m4 make meson ninja optipng pam pam-dev patch pkgconfig pretendroot sed shadow tar uhu-base uhubuild-here uhu-pkg which xz-dev zlib-dev zstd-dev
 
