From e1efbf991b9858f6442faca469ab158914d0ba81 Mon Sep 17 00:00:00 2001
From: rezso <rezso@rezso.net>
Date: Fri, 15 Aug 2025 02:10:09 +0200
Subject: bash 5.3 fix


diff --git a/src/uhubuild-extract b/src/uhubuild-extract
index 32142d9..138c571 100755
--- a/src/uhubuild-extract
+++ b/src/uhubuild-extract
@@ -6,7 +6,7 @@ if [ $# != 1 ]; then
 	die 'uhubuild-extract: Szükségem van argumentumra!'
 fi
 
-if [ "$UID" != "0" ]; then
+if [ "$UID" != 0 ]; then
 	echo 'Figyelem! Nem root vagy!' >&2
 	echo 'A kicsomagolás során elvész a fájlok tulajdonosa és csoportja.' >&2
 	echo 'Ha újra össze szeretnéd csomagolni a fájlokat egy uhu csomaggá,' >&2
diff --git a/src/uhubuild-pack b/src/uhubuild-pack
index 954decd..a009a1d 100755
--- a/src/uhubuild-pack
+++ b/src/uhubuild-pack
@@ -18,7 +18,7 @@ chmod -R u+w DEBIAN
 
 if [ ! -f data.tar.gz ]; then
 
-if [ "$UID" != "0" ]; then
+if [ "$UID" != 0 ]; then
 	echo 'Figyelem! Nem root vagy!' >&2
 	echo 'A készülő csomagban a fájlok tulajdonosa minden bizonnyal nem root lesz.' >&2
 	echo 'Válts át root felhasználóvá, vagy sima felhasználóként használd' >&2
