From 9b91381065b1f535523b8a3373cf127a620492a5 Mon Sep 17 00:00:00 2001
From: rezso <rezso@rezso.net>
Date: Mon, 3 Mar 2025 21:48:59 +0100
Subject: commonenvironment: qt5, qt6 path nem kell


diff --git a/src/uhubuild-commonenvironment b/src/uhubuild-commonenvironment
index 191befe..334d4a3 100644
--- a/src/uhubuild-commonenvironment
+++ b/src/uhubuild-commonenvironment
@@ -19,8 +19,7 @@ for i in /etc/env.d/*.env; do
 done
 unset i
 
-# extra path: qt5,qt6
-export PATH="/usr/lib/uhubuild${UHUBUILD_CCACHE_DIR:+:/usr/lib/ccache}:/sbin:/usr/sbin:/bin:/usr/bin:/usr/lib/qt5/bin:/usr/lib/qt6/bin"
+export PATH="/usr/lib/uhubuild${UHUBUILD_CCACHE_DIR:+:/usr/lib/ccache}:/sbin:/usr/sbin:/bin:/usr/bin"
 
 export LC_ALL=hu_HU.UTF-8
 export HOME=~
