diff -Naur a/src/runtime/kwalletd/CMakeLists.txt b/src/runtime/kwalletd/CMakeLists.txt
--- a/src/runtime/kwalletd/CMakeLists.txt	2025-05-16 12:50:22.000000000 +0200
+++ b/src/runtime/kwalletd/CMakeLists.txt	2025-05-18 00:25:46.576341248 +0200
@@ -11,15 +11,12 @@
 
 
 if(NOT WIN32)
-    configure_file(org.kde.kwalletd5.service.in
-                   ${CMAKE_CURRENT_BINARY_DIR}/org.kde.kwalletd5.service)
     configure_file(org.kde.kwalletd6.service.in
                    ${CMAKE_CURRENT_BINARY_DIR}/org.kde.kwalletd6.service)
 endif()
 
 install(
     FILES
-        ${CMAKE_CURRENT_BINARY_DIR}/org.kde.kwalletd5.service
         ${CMAKE_CURRENT_BINARY_DIR}/org.kde.kwalletd6.service
     DESTINATION ${KDE_INSTALL_DBUSSERVICEDIR}
 )
