diff -Naur a/desktop/cups.desktop.in b/desktop/cups.desktop.in
--- a/desktop/cups.desktop.in	2024-06-18 13:11:05.000000000 +0200
+++ b/desktop/cups.desktop.in	2024-06-20 00:19:11.810116583 +0200
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Categories=System;Printing;HardwareSettings;X-Red-Hat-Base;
-Exec=@CUPS_HTMLVIEW@ http://localhost:631/
+Exec=xdg-open http://localhost:631/
 Icon=cups
 StartupNotify=false
 Terminal=false
