diff -Naur a/servers/slapd/slapd.ldif b/servers/slapd/slapd.ldif
--- a/servers/slapd/slapd.ldif	2024-05-21 19:19:11.000000000 +0200
+++ b/servers/slapd/slapd.ldif	2024-11-03 20:59:16.783564208 +0100
@@ -88,7 +88,7 @@
 # The database directory MUST exist prior to running slapd AND 
 # should only be accessible by the slapd and slap tools.
 # Mode 700 recommended.
-olcDbDirectory:	%LOCALSTATEDIR%/openldap-data
+olcDbDirectory:	%LOCALSTATEDIR%/lib/openldap-data
 # Indices to maintain
 olcDbIndex: objectClass eq
 
