diff -Naur a/glfw/wl_platform.h b/glfw/wl_platform.h
--- a/glfw/wl_platform.h	2022-11-07 11:06:42.000000000 +0100
+++ b/glfw/wl_platform.h	2023-07-30 18:37:14.851135588 +0200
@@ -125,6 +125,7 @@
     TOPLEVEL_STATE_TILED_RIGHT = 32,
     TOPLEVEL_STATE_TILED_TOP = 64,
     TOPLEVEL_STATE_TILED_BOTTOM = 128,
+    TOPLEVEL_STATE_SUSPENDED = 256,
 } WaylandWindowState;
 
 typedef struct glfw_wl_xdg_activation_request {
diff -Naur a/glfw/wl_window.c b/glfw/wl_window.c
--- a/glfw/wl_window.c	2022-11-07 11:06:42.000000000 +0100
+++ b/glfw/wl_window.c	2023-07-30 18:40:11.103709857 +0200
@@ -541,6 +541,7 @@
             C(TOPLEVEL_STATE_TILED_RIGHT);
             C(TOPLEVEL_STATE_TILED_TOP);
             C(TOPLEVEL_STATE_TILED_BOTTOM);
+            C(TOPLEVEL_STATE_SUSPENDED);
 #undef C
         }
     }
