diff -Naur a/cmake/systemd.cmake b/cmake/systemd.cmake
--- a/cmake/systemd.cmake	2023-02-06 20:55:26.000000000 +0100
+++ b/cmake/systemd.cmake	2023-04-07 14:27:04.463009836 +0200
@@ -50,8 +50,7 @@
           SET(SYSTEMD_SCRIPTS ${SYSTEMD_SCRIPTS} galera_new_cluster galera_recovery)
         ENDIF()
         IF(DEB)
-          SET(SYSTEMD_EXECSTARTPRE "ExecStartPre=/usr/bin/install -m 755 -o mysql -g root -d /var/run/mysqld")
-          SET(SYSTEMD_EXECSTARTPOST "ExecStartPost=/etc/mysql/debian-start")
+          SET(SYSTEMD_EXECSTARTPOST "ExecStartPost=/usr/bin/mysqld-ping")
         ENDIF()
 	IF(URING_FOUND)
 	  SET(SYSTEMD_LIMIT "# For liburing and io_uring_setup()
