diff -Naur a/ncftp/ls.c b/ncftp/ls.c
--- a/ncftp/ls.c	2017-12-16 21:47:57.000000000 +0100
+++ b/ncftp/ls.c	2024-06-30 22:12:04.342071188 +0200
@@ -55,7 +55,7 @@
 void InitLsMonths(void)
 {
 	struct tm lt;
-	int i;
+	int i, j, k;
 
 	memset(gLsMon, 0, sizeof(gLsMon));
 	(void) Localtime(0, &lt);
@@ -64,6 +64,13 @@
 	for (i=0; i<12; i++) {
 		lt.tm_mon = i;
 		(void) strftime(gLsMon[i], sizeof(gLsMon[i]) - 1, "%b", &lt);
+		j = k = 0;
+		while (k < 4 && j < sizeof(gLsMon[i])) {
+			unsigned char c = gLsMon[i][j];
+			if (c < 128 || c >= 192) k++;
+			j++;
+		}
+		if (k == 4) gLsMon[i][j-1] = '\0';
 	}
 	(void) strcpy(gLsMon[i], "BUG");
 }	/* InitLsMonths */
