diff -Naur a/lang/calamares_hu.ts b/lang/calamares_hu.ts
--- a/lang/calamares_hu.ts	2024-04-16 15:46:09.000000000 +0200
+++ b/lang/calamares_hu.ts	2024-05-25 20:46:39.000658907 +0200
@@ -3985,7 +3985,7 @@
     <message>
       <location filename="../src/modules/partition/jobs/SetPartitionFlagsJob.cpp" line="149"/>
       <source>The installer failed to set flags on partition %1.</source>
-      <translation>A telepítőnek nem sikerült a zászlók beállítása a partíción %1.</translation>
+      <translation>A telepítőnek nem sikerült a jelzők beállítása a partíción: %1.</translation>
     </message>
   </context>
   <context>
@@ -3993,7 +3993,7 @@
     <message>
       <location filename="../src/modules/users/SetPasswordJob.cpp" line="38"/>
       <source>Set password for user %1</source>
-      <translation>%1 felhasználó jelszó beállítása</translation>
+      <translation>%1 felhasználói jelszó beállítása</translation>
     </message>
     <message>
       <location filename="../src/modules/users/SetPasswordJob.cpp" line="44"/>
