diff -Naur a/src/modules/displaymanager/displaymanager.conf b/src/modules/displaymanager/displaymanager.conf
--- a/src/modules/displaymanager/displaymanager.conf	2024-04-16 15:46:09.000000000 +0200
+++ b/src/modules/displaymanager/displaymanager.conf	2024-05-25 20:48:02.313035964 +0200
@@ -20,9 +20,7 @@
   - sddm
   - lightdm
   - gdm
-  - mdm
   - lxdm
-  - greetd
 
 # Enable the following settings to force a desktop environment
 # in your displaymanager configuration file. This will attempt
diff -Naur a/src/modules/displaymanager/main.py b/src/modules/displaymanager/main.py
--- a/src/modules/displaymanager/main.py	2024-04-16 15:46:09.000000000 +0200
+++ b/src/modules/displaymanager/main.py	2024-05-25 20:48:02.317035982 +0200
@@ -199,6 +199,7 @@
     DesktopEnvironment('/usr/bin/cutefish-session', 'cutefish-xsession'),
     DesktopEnvironment('/usr/bin/river', 'river'),
     DesktopEnvironment('/usr/bin/Hyprland', 'hyprland'),
+    DesktopEnvironment('/usr/bin/start-lumina-desktop', 'Lumina-DE'),
 ]
 
 
@@ -963,6 +964,8 @@
             libcalamares.utils.debug("{!s} selected but not installed".format(dm))
             if dm in displaymanagers:
                 displaymanagers.remove(dm)
+        else:
+            libcalamares.utils.target_env_call(['systemctl', '-f', 'enable', dm])
 
     if not dm_impl:
         libcalamares.utils.warning(
