diff -Naur a/src/modules/dracut/main.py b/src/modules/dracut/main.py
--- a/src/modules/dracut/main.py	2024-12-31 21:44:40.000000000 +0100
+++ b/src/modules/dracut/main.py	2025-01-09 23:06:18.293511973 +0100
@@ -36,14 +36,10 @@
     :return:
     """
     # Fetch the job configuration
-    initramfs_name = libcalamares.job.configuration.get('initramfsName', None)
-    dracut_options = libcalamares.job.configuration.get('options', ['-f'])
-
-    if initramfs_name:
-        dracut_options.append(initramfs_name)
+    target_env_process_output(['mkinitrd-installedkernel'])
 
     try:
-        target_env_process_output(['dracut'] + dracut_options)
+        target_env_process_output(['dracut', '-f', '-N'])
     except subprocess.CalledProcessError as cpe:
         libcalamares.utils.warning(f"Dracut failed with output: {cpe.output}")
         return cpe.returncode
