diff -Naur a/src/modules/users/CreateUserJob.cpp b/src/modules/users/CreateUserJob.cpp
--- a/src/modules/users/CreateUserJob.cpp	2023-12-12 00:04:06.000000000 +0100
+++ b/src/modules/users/CreateUserJob.cpp	2023-12-17 00:14:30.721420193 +0100
@@ -60,8 +60,7 @@
     }
 #else
     useraddCommand << "useradd"
-                   << "-m"
-                   << "-U";
+                   << "-m";
     if ( !shell.isEmpty() )
     {
         useraddCommand << "-s" << shell;
@@ -152,7 +151,7 @@
 
     m_status = tr( "Setting file permissions" );
     emit progress( 0.9 );
-    QString userGroup = QString( "%1:%2" ).arg( m_config->loginName() ).arg( m_config->loginName() );
+    QString userGroup = QString( "%1:%2" ).arg( m_config->loginName() ).arg( "users" );
     QString homeDir = QString( "/home/%1" ).arg( m_config->loginName() );
     auto commandResult = Calamares::System::instance()->targetEnvCommand( { "chown", "-R", userGroup, homeDir } );
     if ( commandResult.getExitCode() )
