diff -Naur a/settings.conf b/settings.conf
--- a/settings.conf	2023-12-12 00:04:06.000000000 +0100
+++ b/settings.conf	2023-12-17 00:36:19.903966483 +0100
@@ -122,6 +122,7 @@
   - partition
   - users
 #  - tracking
+  - netinstall
   - summary
 - exec:
 #  - dummycpp
@@ -130,27 +131,30 @@
   - partition
 #  - zfs
   - mount
-  - unpackfs
+# - unpackfs
+  - chrootcfg
+  - networkcfg
+  - packages
   - machineid
   - fstab
   - locale
-  - keyboard
+# - keyboard
   - localecfg
 #  - luksbootkeyfile
 #  - luksopenswaphookcfg
 #  - dracutlukscfg
 #  - plymouthcfg
 #  - zfshostid
-  - initcpiocfg
-  - initcpio
+#  - initcpiocfg
+#  - initcpio
   - users
   - displaymanager
-  - networkcfg
+# - networkcfg
   - hwclock
-  - services-systemd
-#  - dracut
-  - initramfs
-#  - grubcfg
+# - services-systemd
+  - dracut
+# - initramfs
+  - grubcfg
   - bootloader
   - umount
 - show:
@@ -234,3 +238,7 @@
 #
 #
 quit-at-end: false
+
+packageOperations:
+  -install
+  -remove
