diff -Naur a/src/modules/unpackfs/unpackfs.conf b/src/modules/unpackfs/unpackfs.conf
--- a/src/modules/unpackfs/unpackfs.conf	2024-12-31 21:44:40.000000000 +0100
+++ b/src/modules/unpackfs/unpackfs.conf	2025-01-09 23:23:33.260781134 +0100
@@ -116,13 +116,6 @@
 #        optional: true
 
 unpack:
-    -   source: ../CHANGES
-        sourcefs: file
-        destination: "/tmp/changes.txt"
-        weight: 1  # Single file
-    -   source: src/qml/calamares/slideshow
-        sourcefs: file
-        destination: "/tmp/slideshow/"
-        exclude: [ "*.qmlc", "qmldir" ]
-        weight: 5  # Lots of files
-        # excludeFile: /etc/calamares/modules/unpackfs/exclude-list.txt
+    -   source: "/dev/live-base"
+        sourcefs: "squashfs"
+        destination: ""
