diff -Naur a/src/modules/users/users.conf b/src/modules/users/users.conf
--- a/src/modules/users/users.conf	2025-07-21 18:32:10.000000000 +0200
+++ b/src/modules/users/users.conf	2025-08-22 15:09:35.191805694 +0200
@@ -38,13 +38,6 @@
     - name: users
       must_exist: true
       system: true
-    - lp
-    - video
-    - network
-    - storage
-    - name: wheel
-      must_exist: false
-      system: true
     - audio
     - name: nopasswdlogin
       must_exist: false
@@ -59,7 +52,7 @@
 # remove this setting (delete or comment out the line below). Otherwise,
 # the setting will be duplicated in the `/etc/sudoers.d/10-installer` file,
 # potentially confusing users.
-sudoersGroup:    wheel
+# sudoersGroup:    wheel
 
 # Some Distributions require a 'autologin' group for the user.
 # Autologin causes a user to become automatically logged in to
