diff -Naur a/Makefile b/Makefile
--- a/Makefile	2024-02-25 00:11:10.000000000 +0100
+++ b/Makefile	2024-03-30 01:38:30.876913642 +0100
@@ -19,7 +19,7 @@
 DNS=
 
 # Build libpci as a shared library (yes/no; or local for testing; requires GCC)
-SHARED=no
+SHARED=yes
 
 # Use libkmod to resolve kernel modules on Linux (yes/no, default: detect)
 LIBKMOD=
@@ -32,9 +32,9 @@
 ABI_VERSION=3
 
 # Installation directories
-PREFIX=/usr/local
+PREFIX=/usr
 BINDIR=$(PREFIX)/bin
-SBINDIR=$(PREFIX)/sbin
+SBINDIR=$(PREFIX)/bin
 SHAREDIR=$(PREFIX)/share
 IDSDIR=$(SHAREDIR)
 MANDIR:=$(shell if [ -d $(PREFIX)/share/man ] ; then echo $(PREFIX)/share/man ; else echo $(PREFIX)/man ; fi)
