diff -Naur a/pyproject.toml b/pyproject.toml
--- a/pyproject.toml	2020-02-02 01:00:00.000000000 +0100
+++ b/pyproject.toml	2024-03-14 23:37:22.351220654 +0100
@@ -1,5 +1,5 @@
 [build-system]
-requires = ["hatchling", "hatch-vcs", "hatch-fancy-pypi-readme"]
+requires = ["hatchling", "hatch-vcs"]
 build-backend = "hatchling.build"
 
 [tool.hatch.version]
@@ -34,10 +34,8 @@
   "Programming Language :: Python :: 3.12",
   "Programming Language :: Python :: Implementation :: CPython",
   "Programming Language :: Python :: Implementation :: PyPy",
-  "Topic :: File Formats :: JSON",
-  "Topic :: File Formats :: JSON :: JSON Schema",
 ]
-dynamic = ["version", "readme"]
+dynamic = ["version"]
 dependencies = [
   "attrs>=22.2.0",
   "jsonschema-specifications>=2023.03.6",
@@ -82,30 +80,6 @@
 Changelog = "https://github.com/python-jsonschema/jsonschema/blob/main/CHANGELOG.rst"
 Source = "https://github.com/python-jsonschema/jsonschema"
 
-[tool.hatch.metadata.hooks.fancy-pypi-readme]
-content-type = "text/x-rst"
-
-[[tool.hatch.metadata.hooks.fancy-pypi-readme.fragments]]
-path = "README.rst"
-end-before = ".. start cut from PyPI"
-
-[[tool.hatch.metadata.hooks.fancy-pypi-readme.fragments]]
-path = "README.rst"
-start-after = ".. end cut from PyPI\n\n\n"
-
-[[tool.hatch.metadata.hooks.fancy-pypi-readme.fragments]]
-text = """
-
-
-Release Information
--------------------
-
-"""
-
-[[tool.hatch.metadata.hooks.fancy-pypi-readme.fragments]]
-path = "CHANGELOG.rst"
-pattern = "(^v.+?)\nv"
-
 [tool.coverage.html]
 show_contexts = true
 skip_covered = false
