diff -Naur a/lib/socket-connect.c b/lib/socket-connect.c
--- a/lib/socket-connect.c	2022-02-15 07:31:19.000000000 +0100
+++ b/lib/socket-connect.c	2025-04-26 21:23:41.923909426 +0200
@@ -396,7 +396,7 @@
       return -1;
   }
 
-  newfd = accept (*fd, saddr, &request->remote_addr_len);
+  newfd = accept (*fd, saddr, (socklen_t * restrict)&request->remote_addr_len);
   if (newfd == -1)
   {
       request->logging_function (gftp_logging_error, request,
