diff -Naur a/bootStrap.bash b/bootStrap.bash
--- a/bootStrap.bash	2025-01-11 02:27:26.000000000 +0100
+++ b/bootStrap.bash	2025-01-11 02:56:37.468107442 +0100
@@ -88,10 +88,10 @@
     $ASAN \
     $DEBUG \
     -G "$BUILDER" \
-    "$SOURCEDIR" >&/tmp/logCmake$BASE || fail "cmake,result in /tmp/logCmake$BASE"
+    "$SOURCEDIR"
   $ADATE
   echo "   $BASE:Build started..."
-  ${MAKER} >&/tmp/log$BASE || fail "${MAKER}, result in /tmp/log$BASE"
+  ${MAKER}
   if [ "x$PKG" != "x" ]; then
     DESTDIR="${FAKEROOT_DIR}/tmp" $FAKEROOT_COMMAND ${MAKER} package || fail "packaging"
   fi
@@ -296,7 +296,7 @@
 fi
 SRCTOP=$(cd $(dirname "$0") && pwd)
 POSTFIX=""
-FAKEROOT_DIR="${BUILDTOP}/install"
+FAKEROOT_DIR="${UB_INSTALLDIR}"
 if [ "x$external_liba52" = "x1" ]; then
   EXTRA_CMAKE_DEFS="-DUSE_EXTERNAL_LIBA52=true $EXTRA_CMAKE_DEFS"
 fi
