diff -Naur a/avidemux/qt4/xdg_data/org.avidemux.Avidemux.appdata.xml.in b/avidemux/qt4/xdg_data/org.avidemux.Avidemux.appdata.xml.in
--- a/avidemux/qt4/xdg_data/org.avidemux.Avidemux.appdata.xml.in	2025-01-11 02:27:26.000000000 +0100
+++ b/avidemux/qt4/xdg_data/org.avidemux.Avidemux.appdata.xml.in	2025-01-11 02:58:14.776517190 +0100
@@ -20,7 +20,7 @@
     </categories>
     <url type="homepage">https://www.avidemux.org/</url>
     <provides>
-        <binary>avidemux3_qt5</binary>
+        <binary>avidemux3_qt6</binary>
     </provides>
     <update_contact>Avidemux team -- avidemux.org/smif/</update_contact>
     <screenshots>
diff -Naur a/avidemux/qt4/xdg_data/org.avidemux.Avidemux.desktop b/avidemux/qt4/xdg_data/org.avidemux.Avidemux.desktop
--- a/avidemux/qt4/xdg_data/org.avidemux.Avidemux.desktop	2025-01-11 02:27:26.000000000 +0100
+++ b/avidemux/qt4/xdg_data/org.avidemux.Avidemux.desktop	2025-01-11 02:58:33.256594843 +0100
@@ -2,7 +2,7 @@
 Name=Avidemux
 GenericName=Video Editor
 Comment=Multiplatform video editor
-Exec=avidemux3_qt5 %f
+Exec=avidemux3_qt6 %f
 Icon=org.avidemux.Avidemux
 Terminal=false
 Type=Application
