diff -Naur a/startx.cpp b/startx.cpp
--- a/startx.cpp	2025-01-05 19:25:56.000000000 +0100
+++ b/startx.cpp	2025-01-14 23:16:44.710270680 +0100
@@ -281,7 +281,7 @@
     dummy=0
 
     XCOMM create a file with auth information for the server. ':0' is a dummy.
-    xserverauthfile=$HOME/.serverauth.$$
+    xserverauthfile=`mktemp -p /tmp serverauth.XXXXXXXXXX`
     trap "rm -f '$xserverauthfile'" HUP INT QUIT ILL TRAP BUS TERM
     xauth -q -f "$xserverauthfile" << EOF
 add :$dummy . $mcookie
