diff -Naur a/mesonbuild/coredata.py b/mesonbuild/coredata.py
--- a/mesonbuild/coredata.py	2024-07-26 02:23:53.000000000 +0200
+++ b/mesonbuild/coredata.py	2024-08-23 11:48:08.791323845 +0200
@@ -523,7 +523,7 @@
             opt = '2'
             debug = True
         elif value == 'release':
-            opt = '3'
+            opt = '2'
             debug = False
         elif value == 'minsize':
             opt = 's'
@@ -552,7 +552,7 @@
             opt = '2'
             debug = True
         elif value == 'release':
-            opt = '3'
+            opt = '2'
             debug = False
         elif value == 'minsize':
             opt = 's'
diff -Naur a/mesonbuild/options.py b/mesonbuild/options.py
--- a/mesonbuild/options.py	2024-07-26 02:23:53.000000000 +0200
+++ b/mesonbuild/options.py	2024-08-23 11:50:25.695646039 +0200
@@ -428,7 +428,7 @@
     (OptionKey('errorlogs'),       BuiltinOption(UserBooleanOption, "Whether to print the logs from failing tests", True)),
     (OptionKey('install_umask'),   BuiltinOption(UserUmaskOption, 'Default umask to apply on permissions of installed files', '022')),
     (OptionKey('layout'),          BuiltinOption(UserComboOption, 'Build directory layout', 'mirror', choices=['mirror', 'flat'])),
-    (OptionKey('optimization'),    BuiltinOption(UserComboOption, 'Optimization level', '0', choices=['plain', '0', 'g', '1', '2', '3', 's'])),
+    (OptionKey('optimization'),    BuiltinOption(UserComboOption, 'Optimization level', '2', choices=['plain', '0', 'g', '1', '2', '3', 's'])),
     (OptionKey('prefer_static'),   BuiltinOption(UserBooleanOption, 'Whether to try static linking before shared linking', False)),
     (OptionKey('stdsplit'),        BuiltinOption(UserBooleanOption, 'Split stdout and stderr in test logs', True)),
     (OptionKey('strip'),           BuiltinOption(UserBooleanOption, 'Strip targets on install', False)),
