diff -Naur a/mesonbuild/coredata.py b/mesonbuild/coredata.py
--- a/mesonbuild/coredata.py	2025-04-03 19:13:56.000000000 +0200
+++ b/mesonbuild/coredata.py	2025-05-22 16:05:18.833214602 +0200
@@ -529,7 +529,7 @@
             opt = '2'
             debug = True
         elif value == 'release':
-            opt = '3'
+            opt = '2'
             debug = False
         elif value == 'minsize':
             opt = 's'
@@ -558,7 +558,7 @@
             opt = '2'
             debug = True
         elif value == 'release':
-            opt = '3'
+            opt = '2'
             debug = False
         elif value == 'minsize':
             opt = 's'
diff -Naur a/mesonbuild/options.py b/mesonbuild/options.py
--- a/mesonbuild/options.py	2025-04-03 19:13:56.000000000 +0200
+++ b/mesonbuild/options.py	2025-05-22 16:06:27.785475287 +0200
@@ -636,7 +636,7 @@
     (OptionKey('errorlogs'),       BuiltinOption(UserBooleanOption, "Whether to print the logs from failing tests", True)),
     (OptionKey('install_umask'),   BuiltinOption(UserUmaskOption, 'Default umask to apply on permissions of installed files', '022')),
     (OptionKey('layout'),          BuiltinOption(UserComboOption, 'Build directory layout', 'mirror', choices=['mirror', 'flat'])),
-    (OptionKey('optimization'),    BuiltinOption(UserComboOption, 'Optimization level', '0', choices=['plain', '0', 'g', '1', '2', '3', 's'])),
+    (OptionKey('optimization'),    BuiltinOption(UserComboOption, 'Optimization level', '2', choices=['plain', '0', 'g', '1', '2', '3', 's'])),
     (OptionKey('prefer_static'),   BuiltinOption(UserBooleanOption, 'Whether to try static linking before shared linking', False)),
     (OptionKey('stdsplit'),        BuiltinOption(UserBooleanOption, 'Split stdout and stderr in test logs', True)),
     (OptionKey('strip'),           BuiltinOption(UserBooleanOption, 'Strip targets on install', False)),
