diff -Naur a/completions/Makefile.am b/completions/Makefile.am
--- a/completions/Makefile.am	2024-12-25 11:28:39.000000000 +0100
+++ b/completions/Makefile.am	2025-09-01 00:20:13.335349641 +0200
@@ -186,7 +186,6 @@
 		inotifywait \
 		_insmod \
 		installpkg \
-		interdiff \
 		invoke-rc.d \
 		_ionice \
 		ip \
diff -Naur a/completions/Makefile.in b/completions/Makefile.in
--- a/completions/Makefile.in	2024-12-25 11:28:57.000000000 +0100
+++ b/completions/Makefile.in	2025-09-01 00:20:24.883377065 +0200
@@ -429,7 +429,6 @@
 		inotifywait \
 		_insmod \
 		installpkg \
-		interdiff \
 		invoke-rc.d \
 		_ionice \
 		ip \
