diff -Naur a/plugins/dockers/lut/ocio_display_filter_vfx2021.cpp b/plugins/dockers/lut/ocio_display_filter_vfx2021.cpp
--- a/plugins/dockers/lut/ocio_display_filter_vfx2021.cpp	2023-05-26 10:40:18.000000000 +0200
+++ b/plugins/dockers/lut/ocio_display_filter_vfx2021.cpp	2023-09-30 23:21:08.217074118 +0200
@@ -506,8 +506,9 @@
         unsigned width = 0;
         unsigned height = 0;
         OCIO::GpuShaderDesc::TextureType channel = OCIO::GpuShaderDesc::TEXTURE_RGB_CHANNEL;
+        OCIO::GpuShaderCreator::TextureDimensions dimensions;
         OCIO::Interpolation interpolation = OCIO::INTERP_LINEAR;
-        shaderDesc->getTexture(idx, textureName, samplerName, width, height, channel, interpolation);
+        shaderDesc->getTexture(idx, textureName, samplerName, width, height, channel, dimensions, interpolation);
 
         if (!textureName || !*textureName || !samplerName || !*samplerName || width == 0) {
             errOpenGL << "The texture data is corrupted";
