diff -Naur a/src/useradd.c b/src/useradd.c
--- a/src/useradd.c	2025-02-11 20:13:24.186219074 +0100
+++ b/src/useradd.c	2025-02-11 20:31:31.011928122 +0100
@@ -2358,7 +2358,6 @@
 	gid_t         gid;
 	mode_t        mode;
 	const char    *spool;
-	struct group  *gr;
 
 	if (strcasecmp(create_mail_spool, "yes") != 0)
 		return;
@@ -2394,16 +2393,8 @@
 		return;
 	}
 
-	gr = prefix_getgrnam("mail"); /* local, no need for xgetgrnam */
-	if (NULL == gr) {
-		fputs(_("Group 'mail' not found. Creating the user mailbox file with 0600 mode.\n"),
-		       stderr);
-		gid = user_gid;
-		mode = 0600;
-	} else {
-		gid = gr->gr_gid;
-		mode = 0660;
-	}
+	gid = user_gid;
+	mode = 0600;
 
 	if (fchown(fd, user_id, gid) != 0
 	 || fchmod(fd, mode) != 0)
