diff -Naur a/src/useradd.c b/src/useradd.c
--- a/src/useradd.c	2025-04-17 03:01:42.046104190 +0200
+++ b/src/useradd.c	2025-04-17 03:04:09.034626220 +0200
@@ -2312,7 +2312,6 @@
 	gid_t         gid;
 	mode_t        mode;
 	const char    *spool;
-	struct group  *gr;
 
 	if (!strcaseeq(create_mail_spool, "yes"))
 		return;
@@ -2348,16 +2347,8 @@
 		return;
 	}
 
-	gr = prefix_getgrnam("mail"); /* local, no need for xgetgrnam */
-	if (NULL == gr) {
-		fputs(_("Group 'mail' not found. Creating the user mailbox file with 0600 mode.\n"),
-		       stderr);
-		gid = user_gid;
-		mode = 0600;
-	} else {
-		gid = gr->gr_gid;
-		mode = 0660;
-	}
+	gid = user_gid;
+	mode = 0600;
 
 	if (fchown(fd, user_id, gid) != 0
 	 || fchmod(fd, mode) != 0)
