diff -Naur a/etc/pam.d/login b/etc/pam.d/login
--- a/etc/pam.d/login	2019-07-23 17:26:08.000000000 +0200
+++ b/etc/pam.d/login	2024-02-02 19:34:21.241165383 +0100
@@ -4,8 +4,10 @@
 account		required	pam_nologin.so
 account		include		system-auth
 password	include		system-auth
-session		required	pam_selinux.so close
 session		include		system-auth
 session		required	pam_loginuid.so
 session		optional	pam_console.so
-session		required	pam_selinux.so open
+
+# Set default environment variables for the user
+session   required    pam_env.so
+session   required    pam_envfeed.so
diff -Naur a/etc/pam.d/su b/etc/pam.d/su
--- a/etc/pam.d/su	2019-07-23 17:26:08.000000000 +0200
+++ b/etc/pam.d/su	2024-02-02 19:35:57.137308838 +0100
@@ -1,13 +1,11 @@
 #%PAM-1.0
 auth		sufficient	pam_rootok.so
-# Uncomment the following line to implicitly trust users in the "wheel" group.
-#auth		sufficient	pam_wheel.so trust use_uid
-# Uncomment the following line to require a user to be in the "wheel" group.
-auth		required	pam_wheel.so use_uid
 auth		include		system-auth
 account		include		system-auth
 password	include		system-auth
-session		required	pam_selinux.so close
 session		include		system-auth
-session		required	pam_selinux.so open multiple
 session		optional	pam_xauth.so
+
+# Set default environment variables for the user
+session   required    pam_env.so
+session   required    pam_envfeed.so
