diff -Naur a/src/groupadd.c b/src/groupadd.c
--- a/src/groupadd.c	2025-03-19 19:57:26.000000000 +0100
+++ b/src/groupadd.c	2025-04-17 03:14:07.988235954 +0200
@@ -112,6 +112,7 @@
 	(void) fputs (_("  -o, --non-unique              allow to create groups with duplicate\n"
 	                "                                (non-unique) GID\n"), usageout);
 	(void) fputs (_("  -p, --password PASSWORD       use this encrypted password for the new group\n"), usageout);
+	(void) fputs (_("  -q, --quiet                   succeed quietly if user+uid already exists\n"), usageout);
 	(void) fputs (_("  -r, --system                  create a system account\n"), usageout);
 	(void) fputs (_("  -R, --root CHROOT_DIR         directory to chroot into\n"), usageout);
 	(void) fputs (_("  -P, --prefix PREFIX_DIR       directory prefix\n"), usageout);
@@ -394,6 +395,7 @@
 		{"key",        required_argument, NULL, 'K'},
 		{"non-unique", no_argument,       NULL, 'o'},
 		{"password",   required_argument, NULL, 'p'},
+		{"quiet",      no_argument,       NULL, 'q'},
 		{"system",     no_argument,       NULL, 'r'},
 		{"root",       required_argument, NULL, 'R'},
 		{"prefix",     required_argument, NULL, 'P'},
@@ -401,7 +403,7 @@
 		{NULL, 0, NULL, '\0'}
 	};
 
-	while ((c = getopt_long (argc, argv, "fg:hK:op:rR:P:U:",
+	while ((c = getopt_long (argc, argv, "fg:hK:op:qrR:P:U:",
 		                 long_options, NULL)) != -1) {
 		switch (c) {
 		case 'f':
@@ -451,6 +453,9 @@
 			pflg = true;
 			group_passwd = optarg;
 			break;
+		case 'q': 	/* set 'force' flag for quiet mode */
+			fflg = true;
+			break;
 		case 'r':
 			rflg = true;
 			break;
diff -Naur a/src/groupdel.c b/src/groupdel.c
--- a/src/groupdel.c	2025-03-19 19:57:26.000000000 +0100
+++ b/src/groupdel.c	2025-04-17 03:14:07.992235965 +0200
@@ -305,17 +305,20 @@
 	static struct option long_options[] = {
 		{"help", no_argument,       NULL, 'h'},
 		{"force", no_argument,      NULL, 'f'},
+		{"quiet", no_argument,      NULL, 'q'},
 		{"root", required_argument, NULL, 'R'},
 		{"prefix", required_argument, NULL, 'P'},
 		{NULL, 0, NULL, '\0'}
 	};
 
-	while ((c = getopt_long (argc, argv, "hfR:P:",
+	while ((c = getopt_long (argc, argv, "hfqR:P:",
 	                         long_options, NULL)) != -1) {
 		switch (c) {
 		case 'h':
 			usage (E_SUCCESS);
 			/*@notreached@*/break;
+		case 'q': /* no operation, just for backward compatibility with old UHU patch */
+			break;
 		case 'R': /* no-op, handled in process_root_flag () */
 			break;
 		case 'P': /* no-op, handled in process_prefix_flag () */
diff -Naur a/src/userdel.c b/src/userdel.c
--- a/src/userdel.c	2025-03-19 19:57:26.000000000 +0100
+++ b/src/userdel.c	2025-04-17 03:14:07.992235965 +0200
@@ -925,6 +925,7 @@
 		static struct option long_options[] = {
 			{"force",        no_argument,       NULL, 'f'},
 			{"help",         no_argument,       NULL, 'h'},
+			{"quiet",        no_argument, 		NULL, 'q'},
 			{"remove",       no_argument,       NULL, 'r'},
 			{"root",         required_argument, NULL, 'R'},
 			{"prefix",       required_argument, NULL, 'P'},
@@ -935,9 +936,9 @@
 		};
 		while ((c = getopt_long (argc, argv,
 #ifdef WITH_SELINUX
-		                         "fhrR:P:Z",
+		                         "fhqrR:P:Z",
 #else				/* !WITH_SELINUX */
-		                         "fhrR:P:",
+		                         "fhqrR:P:",
 #endif				/* !WITH_SELINUX */
 		                         long_options, NULL)) != -1) {
 			switch (c) {
@@ -947,6 +948,9 @@
 			case 'h':
 				usage (E_SUCCESS);
 				break;
+			case 'q':  /* set 'force' flag for backward compatibility with old UHU patch */
+				fflg = true;
+				break;
 			case 'r':	/* remove home dir and mailbox */
 				rflg = true;
 				break;
