diff -Naur a/data/org.gnome.SystemMonitor.desktop.in.in b/data/org.gnome.SystemMonitor.desktop.in.in
--- a/data/org.gnome.SystemMonitor.desktop.in.in	2024-03-16 20:12:56.000000000 +0100
+++ b/data/org.gnome.SystemMonitor.desktop.in.in	2024-03-19 02:23:54.038841987 +0100
@@ -1,5 +1,5 @@
 [Desktop Entry]
-Name=System Monitor
+Name=GNOME System Monitor
 Comment=View current processes and monitor system state
 TryExec=gnome-system-monitor
 Exec=gnome-system-monitor
