diff -Naur a/meson.build b/meson.build
--- a/meson.build	2025-03-15 19:50:01.000000000 +0100
+++ b/meson.build	2025-03-27 00:29:28.559431458 +0100
@@ -51,7 +51,6 @@
 ################################################################################
 # Dependencies
 
-catch2     = dependency('catch2-with-main')
 giomm      = dependency('giomm-2.68',  version: '>=2.68')
 glib       = dependency('glib-2.0',    version: '>=2.56.0')
 glibmm     = dependency('glibmm-2.68', version: '>=2.68')
diff -Naur a/src/meson.build b/src/meson.build
--- a/src/meson.build	2025-03-15 19:50:01.000000000 +0100
+++ b/src/meson.build	2025-03-27 00:30:16.251610156 +0100
@@ -149,18 +149,6 @@
   install: true,
 )
 
-unit_tests = executable('unit_tests',
-  [
-    'test/cgroups.cpp',
-    'test/join.cpp'
-  ],
-  dependencies: [
-    catch2,
-    libgsm_dep,
-  ],
-)
-test('unit_tests', unit_tests)
-
 gnome.post_install(
   glib_compile_schemas: true,
 )
