diff -Naur a/wxVillaLib/PropDlg.cpp b/wxVillaLib/PropDlg.cpp
--- a/wxVillaLib/PropDlg.cpp	2021-11-07 09:42:36.000000000 +0100
+++ b/wxVillaLib/PropDlg.cpp	2025-03-30 01:45:32.542474374 +0100
@@ -12,7 +12,9 @@
 #include "utils.h"
 #include <wx/fontdlg.h>
 #include <wx/colordlg.h>
+#ifdef __WXMSW__
 #include <wx/generic/colrdlgg.h>
+#endif
 #include <wx/filedlg.h>
 #include <wx/dirdlg.h>
 #include <wx/grid.h>
