diff -Naur a/src/shared/install.c b/src/shared/install.c
--- a/src/shared/install.c	2022-10-31 19:59:18.000000000 +0100
+++ b/src/shared/install.c	2024-08-22 21:36:05.792567129 +0200
@@ -339,7 +339,8 @@
         assert(verb || r >= 0);
 
         for (size_t i = 0; i < n_changes; i++) {
-                assert(verb || changes[i].type >= 0);
+                if (changes[i].type < 0)
+                        assert(verb);
 
                 /* When making changes here, make sure to also change install_error() in dbus-manager.c. */
 
