diff -Naur a/man/systemd-system.conf.xml b/man/systemd-system.conf.xml
--- a/man/systemd-system.conf.xml	2022-10-31 19:59:18.000000000 +0100
+++ b/man/systemd-system.conf.xml	2023-08-22 15:23:17.277367293 +0200
@@ -364,7 +364,7 @@
         <varname>TimeoutSec=</varname>
         value. <varname>DefaultTimeoutStartSec=</varname> and
         <varname>DefaultTimeoutStopSec=</varname> default to
-        90s. <varname>DefaultTimeoutAbortSec=</varname> is not set by default
+        45s. <varname>DefaultTimeoutAbortSec=</varname> is not set by default
         so that all units fall back to <varname>TimeoutStopSec=</varname>.
         <varname>DefaultRestartSec=</varname> defaults to
         100ms.</para></listitem>
diff -Naur a/src/basic/def.h b/src/basic/def.h
--- a/src/basic/def.h	2022-10-31 19:59:18.000000000 +0100
+++ b/src/basic/def.h	2023-08-22 15:24:32.941602669 +0200
@@ -1,7 +1,7 @@
 /* SPDX-License-Identifier: LGPL-2.1-or-later */
 #pragma once
 
-#define DEFAULT_TIMEOUT_USEC (90*USEC_PER_SEC)
+#define DEFAULT_TIMEOUT_USEC (45*USEC_PER_SEC)
 #define DEFAULT_RESTART_USEC (100*USEC_PER_MSEC)
 #define DEFAULT_CONFIRM_USEC (30*USEC_PER_SEC)
 
diff -Naur a/src/core/system.conf.in b/src/core/system.conf.in
--- a/src/core/system.conf.in	2022-10-31 19:59:18.000000000 +0100
+++ b/src/core/system.conf.in	2023-08-22 15:37:20.948010622 +0200
@@ -43,8 +43,8 @@
 #DefaultTimerAccuracySec=1min
 #DefaultStandardOutput=journal
 #DefaultStandardError=inherit
-#DefaultTimeoutStartSec=90s
-#DefaultTimeoutStopSec=90s
+#DefaultTimeoutStartSec=45s
+#DefaultTimeoutStopSec=45s
 #DefaultTimeoutAbortSec=
 #DefaultDeviceTimeoutSec=90s
 #DefaultRestartSec=100ms
diff -Naur a/src/core/user.conf.in b/src/core/user.conf.in
--- a/src/core/user.conf.in	2022-10-31 19:59:18.000000000 +0100
+++ b/src/core/user.conf.in	2023-08-22 15:37:26.380027758 +0200
@@ -24,8 +24,8 @@
 #DefaultTimerAccuracySec=1min
 #DefaultStandardOutput=inherit
 #DefaultStandardError=inherit
-#DefaultTimeoutStartSec=90s
-#DefaultTimeoutStopSec=90s
+#DefaultTimeoutStartSec=45s
+#DefaultTimeoutStopSec=45s
 #DefaultTimeoutAbortSec=
 #DefaultRestartSec=100ms
 #DefaultStartLimitIntervalSec=10s
diff -Naur a/units/user@.service.in b/units/user@.service.in
--- a/units/user@.service.in	2022-10-31 19:59:18.000000000 +0100
+++ b/units/user@.service.in	2023-08-22 15:27:32.386162307 +0200
@@ -23,6 +23,6 @@
 KillMode=mixed
 Delegate=pids memory cpu
 TasksMax=infinity
-TimeoutStopSec=120s
+TimeoutStopSec=45s
 KeyringMode=inherit
 OOMScoreAdjust=100
