diff -Naur a/lib/lyx.desktop.in b/lib/lyx.desktop.in
--- a/lib/lyx.desktop.in	2024-07-02 21:17:53.000000000 +0200
+++ b/lib/lyx.desktop.in	2024-07-14 14:07:20.262095016 +0200
@@ -4,6 +4,7 @@
 Name=LyX@program_suffix@
 GenericName=Document Processor
 Comment=High level LaTeX frontend
+Comment[hu]=LaTeX fejlesztői környezet
 Keywords=WYSIWYG;WYSIWYM;TeX;LaTeX;GUI;frontend;editor;
 Exec=lyx@program_suffix@ %F
 TryExec=lyx@program_suffix@
