diff -Naur a/meson.build b/meson.build
--- a/meson.build	2025-09-12 11:09:47.000000000 +0200
+++ b/meson.build	2025-09-15 12:12:56.210789040 +0200
@@ -42,7 +42,7 @@
 gweather = dependency('gweather4', version: '>= 3.90.0')
 geocodeglib = dependency('geocode-glib-2.0', version: '>= 3.15.2')
 libportal = dependency('libportal')
-shumate = dependency('shumate-1.0', version: '>= 1.5.alpha')
+shumate = dependency('shumate-1.0', version: '>= 1.4.0')
 
 cc = meson.get_compiler('c')
 
