diff -Naur a/chrome/browser/ui/startup/infobar_utils.cc b/chrome/browser/ui/startup/infobar_utils.cc
--- a/chrome/browser/ui/startup/infobar_utils.cc	2025-04-06 21:57:16.376613904 +0200
+++ b/chrome/browser/ui/startup/infobar_utils.cc	2025-04-06 21:58:45.885086466 +0200
@@ -161,15 +161,5 @@
         ObsoleteSystemInfoBarDelegate::Create(infobar_manager);
       }
     }
-
-#if !BUILDFLAG(IS_CHROMEOS) && !BUILDFLAG(IS_ANDROID)
-    if (!is_web_app &&
-        !startup_command_line.HasSwitch(switches::kNoDefaultBrowserCheck)) {
-      // The default browser prompt should only be shown after the first run.
-      if (is_first_run == chrome::startup::IsFirstRun::kNo) {
-        ShowDefaultBrowserPrompt(profile);
-      }
-    }
-#endif
   }
 }
