diff -Naur a/build/config/compiler/pgo/pgo.gni b/build/config/compiler/pgo/pgo.gni
--- a/build/config/compiler/pgo/pgo.gni	2025-07-28 22:04:51.000000000 +0200
+++ b/build/config/compiler/pgo/pgo.gni	2025-08-04 23:43:29.642517635 +0200
@@ -24,15 +24,6 @@
   # to think some code is hotter than it actually is, potentially causing very
   # bad compile times.
   chrome_pgo_phase = 0
-  if (!is_cronet_build && !dcheck_always_on && is_official_build &&
-      # TODO(crbug.com/1336055): Update this now-outdated condition with regard
-      # to chromecast and determine whether chromeos_is_browser_only is
-      # obsolete.
-      (is_high_end_android || is_win || is_mac || is_fuchsia ||
-       (is_linux && !is_castos) ||
-       (is_ios && use_blink && target_environment == "device"))) {
-    chrome_pgo_phase = 2
-  }
 
   # When using chrome_pgo_phase = 2, read profile data from this path.
   pgo_data_path = ""
