diff -Naur a/src/lib/animation.vala b/src/lib/animation.vala
--- a/src/lib/animation.vala	2024-06-22 02:44:43.000000000 +0200
+++ b/src/lib/animation.vala	2025-03-28 11:26:22.161109682 +0100
@@ -27,7 +27,6 @@
 	* Utility to struct to enable easier animations
 	* Inspired by Clutter.
 	*/
-	[Compact]
 	public class Animation : GLib.Object {
 		public int64 start_time; /**<Start time (microseconds) of animation */
 		public int64 length; /**<Length of animation in microseconds */
diff -Naur a/src/panel/applets/icon-tasklist/animation.vala b/src/panel/applets/icon-tasklist/animation.vala
--- a/src/panel/applets/icon-tasklist/animation.vala	2024-06-22 02:44:43.000000000 +0200
+++ b/src/panel/applets/icon-tasklist/animation.vala	2025-03-28 11:26:51.833231119 +0100
@@ -27,7 +27,6 @@
 	* Utility to struct to enable easier animations
 	* Inspired by Clutter.
 	*/
-	[Compact]
 	public class Animation : GLib.Object {
 		public int64 start_time; /**<Start time (microseconds) of animation */
 		public int64 length; /**<Length of animation in microseconds */
diff -Naur a/src/panel/manager.vala b/src/panel/manager.vala
--- a/src/panel/manager.vala	2024-06-22 02:44:43.000000000 +0200
+++ b/src/panel/manager.vala	2025-03-28 11:26:22.161109682 +0100
@@ -23,7 +23,6 @@
 	/**
 	* Available slots
 	*/
-	[Compact]
 	class Screen : GLib.Object {
 		public PanelPosition slots;
 		public Gdk.Rectangle area;
diff -Naur a/src/panel/settings/themes.vala b/src/panel/settings/themes.vala
--- a/src/panel/settings/themes.vala	2024-06-22 02:44:43.000000000 +0200
+++ b/src/panel/settings/themes.vala	2025-03-28 11:26:22.161109682 +0100
@@ -15,7 +15,6 @@
 	CURSOR_THEME
 }
 
-[Compact]
 class ThemeInfo : GLib.Object {
 	private ThemeType theme_type;
 	private List<string> paths;
