diff -Naur a/methods/gpgv.cc b/methods/gpgv.cc
--- a/methods/gpgv.cc	2024-11-22 17:17:41.000000000 +0100
+++ b/methods/gpgv.cc	2025-05-12 20:46:23.007550675 +0200
@@ -70,13 +70,13 @@
 
 static constexpr Digest Digests[] = {
    {Digest::State::Untrusted, "Invalid digest"},
-   {Digest::State::Untrusted, "MD5"},
-   {Digest::State::Untrusted, "SHA1"},
    {Digest::State::Untrusted, "RIPE-MD/160"},
    {Digest::State::Trusted, "Reserved digest"},
    {Digest::State::Trusted, "Reserved digest"},
    {Digest::State::Trusted, "Reserved digest"},
    {Digest::State::Trusted, "Reserved digest"},
+   {Digest::State::Trusted, "MD5"},
+   {Digest::State::Trusted, "SHA1"},
    {Digest::State::Trusted, "SHA256"},
    {Digest::State::Trusted, "SHA384"},
    {Digest::State::Trusted, "SHA512"},
