diff -Naur a/ftparchive/writer.cc b/ftparchive/writer.cc
--- a/ftparchive/writer.cc	2024-11-22 17:17:41.000000000 +0100
+++ b/ftparchive/writer.cc	2025-05-12 20:47:31.859850737 +0200
@@ -366,7 +366,7 @@
       string const &Arch, bool const IncludeArchAll) :
    FTWScanner(GivenOutput, Arch, IncludeArchAll), Db(DB), Stats(Db.Stats), TransWriter(transWriter)
 {
-   SetExts(".deb .ddeb .udeb");
+   SetExts(".uhu .deb .ddeb .udeb");
    DeLinkLimit = 0;
 
    // Process the command line options
