diff -Naur a/cmdline/apt-key.in b/cmdline/apt-key.in
--- a/cmdline/apt-key.in	2024-11-22 17:17:41.000000000 +0100
+++ b/cmdline/apt-key.in	2025-05-12 20:49:56.868493072 +0200
@@ -731,9 +731,9 @@
 	return
     fi
     # (Maintainer) scripts should not be using apt-key
-    if [ -n "$DPKG_MAINTSCRIPT_PACKAGE" ]; then
-	echo >&2 "Warning: apt-key should not be used in scripts (called from $DPKG_MAINTSCRIPT_NAME maintainerscript of the package ${DPKG_MAINTSCRIPT_PACKAGE})"
-    fi
+    # if [ -n "$DPKG_MAINTSCRIPT_PACKAGE" ]; then
+	# echo >&2 "Warning: apt-key should not be used in scripts (called from $DPKG_MAINTSCRIPT_NAME maintainerscript of the package ${DPKG_MAINTSCRIPT_PACKAGE})"
+    # fi
 
     echo >&2 "Warning: apt-key is deprecated. Manage keyring files in trusted.gpg.d instead (see apt-key(8))."
 }
