diff -Naur a/autogen.sh b/autogen.sh
--- a/autogen.sh	2024-10-23 03:07:24.000000000 +0200
+++ b/autogen.sh	2024-11-10 15:37:30.984526059 +0100
@@ -33,11 +33,6 @@
     exit 1
 }
 
-(test $(grep -q "^GTK_DOC_CHECK" configure.ac)) || {
-    WANT_GTK_DOC=1
-    FEDORA_PKG2="$FEDORA_PKG2 gtk-doc"
-}
-
 (test -f ChangeLog) || {
     touch ChangeLog
 }
@@ -60,11 +55,8 @@
     echo "*** If you wish to pass any to it, please specify them on the" >&2
     echo "*** '$0' command line." >&2
     echo "" >&2
-    (test $WANT_GTK_DOC -eq 1) && CONFIGFLAGS="--enable-gtk-doc $@"
 }
 
-(test $WANT_GTK_DOC -eq 1) && gtkdocize --copy
-
 ACLOCAL_FLAGS="$ACLOCAL_FLAGS -I m4" REQUIRED_AUTOMAKE_VERSION=1.11 \
 autoreconf --verbose --force --install || exit 1
 
diff -Naur a/configure.ac b/configure.ac
--- a/configure.ac	2024-10-23 03:07:24.000000000 +0200
+++ b/configure.ac	2024-11-10 15:43:41.248940947 +0100
@@ -401,12 +401,6 @@
 # Check vapigen.
 VAPIGEN_CHECK([0.16])
 
-# Check for gtk-doc.
-GTK_DOC_CHECK(1.9)
-if test x"$enable_gtk_doc" = x"no"; then
-    enable_gtk_doc="no (disabled, use --enable-gtk-doc to enable)"
-fi
-
 # Check for dbus.
 PKG_CHECK_MODULES(DBUS, [
     dbus-1
@@ -899,10 +893,6 @@
 data/dconf/Makefile
 data/dconf/00-upstream-settings.5
 data/dconf/ibus.5
-docs/Makefile
-docs/reference/Makefile
-docs/reference/ibus/ibus-docs.sgml
-docs/reference/ibus/Makefile
 engine/Makefile
 ibus/_config.py
 ibus/Makefile
diff -Naur a/Makefile.am b/Makefile.am
--- a/Makefile.am	2024-10-23 03:07:24.000000000 +0200
+++ b/Makefile.am	2024-11-10 15:42:58.504892091 +0100
@@ -61,7 +61,6 @@
 	$(DAEMON_DIR) \
 	$(PYTHON_LIB_DIRS) \
 	$(SETUP_DIR) \
-	docs \
 	$(NULL)
 
 ACLOCAL_AMFLAGS = -I m4
