diff -Naur a/gcc/c-family/c.opt b/gcc/c-family/c.opt
--- a/gcc/c-family/c.opt	2024-08-01 10:17:14.000000000 +0200
+++ b/gcc/c-family/c.opt	2025-05-07 00:25:42.597549183 +0200
@@ -1473,7 +1473,7 @@
 Warn when a const variable is unused.
 
 Wunused-local-typedefs
-C ObjC C++ ObjC++ Var(warn_unused_local_typedefs) Warning EnabledBy(Wunused)
+C ObjC C++ ObjC++ Var(warn_unused_local_typedefs) Warning
 Warn when typedefs locally defined in a function are not used.
 
 Wunused-macros
