diff -Naur a/pyproject.toml b/pyproject.toml
--- a/pyproject.toml	2025-05-06 18:14:00.000000000 +0200
+++ b/pyproject.toml	2025-08-01 06:38:25.113641263 +0200
@@ -1,5 +1,5 @@
 [build-system]
-requires = ["setuptools >= 61.0", "setuptools-scm>=8"]
+requires = ["setuptools >= 61.0"]
 build-backend = "setuptools.build_meta"
 
 [project]
@@ -67,8 +67,6 @@
     "tools/dumppdf.py",
 ]
 
-[tool.setuptools_scm]
-
 [tool.ruff.lint]
 extend-select = [
     "W",  # pycodestyle warnings
