diff -Naur a/src/journal/journald.conf b/src/journal/journald.conf
--- a/src/journal/journald.conf	2025-03-05 16:13:29.000000000 +0100
+++ b/src/journal/journald.conf	2025-06-24 21:22:33.633621748 +0200
@@ -24,13 +24,13 @@
 #SyncIntervalSec=5m
 #RateLimitIntervalSec=30s
 #RateLimitBurst=10000
-#SystemMaxUse=
+SystemMaxUse=50M
 #SystemKeepFree=
-#SystemMaxFileSize=
+SystemMaxFileSize=50M
 #SystemMaxFiles=100
-#RuntimeMaxUse=
+RuntimeMaxUse=50M
 #RuntimeKeepFree=
-#RuntimeMaxFileSize=
+RuntimeMaxFileSize=50M
 #RuntimeMaxFiles=100
 #MaxRetentionSec=0
 #MaxFileSec=1month
