diff -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2023-10-30 18:09:56.000000000 +0100
+++ b/CMakeLists.txt	2024-03-22 23:26:55.888708128 +0100
@@ -142,15 +142,8 @@
 
 # install the bin
 install(TARGETS multitail DESTINATION bin)
-# install the config file
-install(FILES multitail.conf DESTINATION etc RENAME multitail.conf.new)
 # install the manual files
 install(FILES multitail.1 DESTINATION share/man/man1)
-# install doc files
-install(FILES manual.html DESTINATION share/doc/multitail-${VERSION})
-install(FILES LICENSE DESTINATION share/doc/multitail-${VERSION})
-install(FILES readme.txt DESTINATION share/doc/multitail-${VERSION})
-install(FILES thanks.txt DESTINATION share/doc/multitail-${VERSION})
 # cp conversion-scripts/* etc/multitail/
 install(DIRECTORY conversion-scripts DESTINATION etc/multitail)
 
