diff -Naur a/qtwebengine/src/3rdparty/chromium/ui/gl/gl_display.cc b/qtwebengine/src/3rdparty/chromium/ui/gl/gl_display.cc
--- a/qtwebengine/src/3rdparty/chromium/ui/gl/gl_display.cc	2024-03-22 21:13:57.000000000 +0100
+++ b/qtwebengine/src/3rdparty/chromium/ui/gl/gl_display.cc	2024-04-06 03:00:24.450429364 +0200
@@ -28,6 +28,10 @@
 #include "ui/gl/gl_implementation.h"
 #include "ui/gl/gl_surface.h"
 
+#if defined(USE_GLX)
+#include "ui/gfx/x/connection.h"
+#endif
+
 #if BUILDFLAG(IS_OZONE)
 #include "ui/ozone/buildflags.h"
 #endif  // BUILDFLAG(IS_OZONE)
