diff -Naur a/Makefile b/Makefile
--- a/Makefile	2021-02-10 18:54:01.000000000 +0100
+++ b/Makefile	2024-04-26 13:37:58.228228751 +0200
@@ -9,9 +9,9 @@
 # Check SVG support with '$ ldd /usr/bin/openbox | grep svg', librsvg must appear..
 # CFLAGS+=-DWITH_SVG
 
-prefix= /usr/local
-DESTDIR ?= $(prefix)
-BINDIR= ${DESTDIR}/bin
+prefix= /usr
+DESTDIR ?=
+BINDIR= ${DESTDIR}/$(prefix)/bin
 
 SRC= $(shell ls src/*.c 2> /dev/null)
 OBJ= $(SRC:.c=.o)
