diff -Naur a/configure b/configure
--- a/configure	2025-04-28 00:48:31.000000000 +0200
+++ b/configure	2025-04-28 00:58:38.210809195 +0200
@@ -6737,9 +6737,6 @@
 fi
 
 
-printf "%s\n" "#define HAVE_LIBGIMP_GIMP_H 1" >>confdefs.h
-
-
 BINPROGS=${BINPROGS}" "${GUIPROGS}
 
 
diff -Naur a/configure.in b/configure.in
--- a/configure.in	2025-04-28 00:48:31.000000000 +0200
+++ b/configure.in	2025-04-28 00:57:31.846608973 +0200
@@ -145,8 +145,6 @@
   fi
 fi
 
-AC_DEFINE([HAVE_LIBGIMP_GIMP_H], 1, [Define to 1 if you have the <libgimp/gimp.h> header file.])
-
 BINPROGS=${BINPROGS}" "${GUIPROGS}
 
 AC_SUBST(INCLUDES)
